

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Quick Start &mdash; dbCAN-new-README 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=87f11076"></script>
      <script src="_static/doctools.js?v=888ff710"></script>
      <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Process the input fasta file" href="input_fasta_data_processing.html" />
    <link rel="prev" title="Database Installation Command" href="prepare_the_database.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            dbCAN-new-README
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="prepare_the_database.html">Database Installation Command</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Quick Start</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#running-example-data-for-cazyme-annotation">1. Running Example Data for CAZyme Annotation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#understanding-the-output">2. Understanding the Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-example-data-for-cgc-annotation-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues">3. Running Example Data for CGC Annotation (please check the previous step for downloading example fasta data, we don’t repeat it here to avoid issues.)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1">4. Understanding the Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-example-data-for-substrate-prediction-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues-here-we-download-the-gff-files">5. Running Example Data for Substrate Prediction (please check the previous step for downloading example fasta data, we don’t repeat it here to avoid issues. Here we download the gff files.)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id2">1. Understanding the Output</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="input_fasta_data_processing.html">Process the input fasta file</a></li>
<li class="toctree-l1"><a class="reference internal" href="CAZyme_annotation.html">CAZyme Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_information_generation.html">Prepare the CGC annotation information</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_annotation.html">Run with CGCFinder</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_CGC_substrate.html">Substrate prediction with CGCs</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_substrate_plots.html">CGC substrate visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads.html">Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_am.html">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_pr.html">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_wk.html">Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_em.html">Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_co_assem.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_subsample.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_assem_free.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">dbCAN-new-README</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Quick Start</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/quick_start.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="quick-start">
<h1>Quick Start<a class="headerlink" href="#quick-start" title="Permalink to this heading"></a></h1>
<p>This section provides a quick guide to running the run_dbcan tool suite with example data and explains the output files generated.</p>
<p>For the updated run_dbcan, we provide two types of approach for users:</p>
<p>1.Automated analysis that can be done with one line of command.</p>
<p>2.One by one command that allows checking problems by breaking down the steps or making autonomous changes to some of the results.</p>
<p>Here we show all the steps in one line of command.</p>
<p>We provide multiple example data sets for users to test the tool suite. The example data sets are available in the <cite>example_data</cite> directory.</p>
<p>We currently support configuration using anaconda and pip.
prepare the conda environment (available at <a class="reference external" href="https://github.com/bcb-unl/run_dbcan_new/tree/master/envs">https://github.com/bcb-unl/run_dbcan_new/tree/master/envs</a>):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yml
conda<span class="w"> </span>activate<span class="w"> </span>run_dbcan_env
</pre></div>
</div>
<section id="running-example-data-for-cazyme-annotation">
<h2>1. Running Example Data for CAZyme Annotation<a class="headerlink" href="#running-example-data-for-cazyme-annotation" title="Permalink to this heading"></a></h2>
<p>To run the dbCAN tool suite on the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/U00096.2">Escherichia coli Strain MG1655</a> example data, use the following command. The input file <cite>EscheriaColiK12MG1655.fna</cite> represents the FASTA format complete genome DNA sequence.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/NCBI_prok_test/EscheriaColiK12MG1655.fna<span class="w"> </span>-O<span class="w"> </span>EscheriaColiK12MG1655.fna
run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.fna<span class="w"> </span>--mode<span class="w"> </span>prok<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_fna<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>For the protein sequence input, use the following command (Please note that the input format is needed for the protein sequence only. <cite>NCBI</cite> represents the fasta ID format like NCBI “&gt;WP_000002088.1”, and the <cite>JGI</cite> mode represents the fasta ID format like JGI “&gt;jgi|Xylhe1|242238|”).:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/NCBI_prok_test/EscheriaColiK12MG1655.faa<span class="w"> </span>-O<span class="w"> </span>EscheriaColiK12MG1655.faa

run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_faa<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI
</pre></div>
</div>
<p>We also provide eukaryotes example data sets. For example, to run the dbCAN tool suite on the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_001619985.1/">Xylona heveae TC161</a> example data, use the following command:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/NCBI_euk_test/Xylona_heveae_TC161.faa<span class="w"> </span>-O<span class="w"> </span>Xylona_heveae_TC161.faa
run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylona_heveae_TC161.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylona_heveae_TC161_faa<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>NCBI
</pre></div>
</div>
<p>And JGI dataset <a class="reference external" href="https://mycocosm.jgi.doe.gov/Xylhe1/Xylhe1.home.html">Xylhe1</a>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/JGI_test/Xylhe1_GeneCatalog_proteins_20130827.aa.fasta<span class="w"> </span>-O<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.aa.fasta
run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.aa.fasta<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylhe1_faa<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>JGI
</pre></div>
</div>
</section>
<section id="understanding-the-output">
<h2>2. Understanding the Output<a class="headerlink" href="#understanding-the-output" title="Permalink to this heading"></a></h2>
<p>After running the tool, several output files are generated in the output folder, each with specific information:</p>
<dl class="simple">
<dt><strong>uniInput.faa</strong></dt><dd><p>The unified input file for subsequent tools, created by Prodigal if a nucleotide sequence is used, or provided by the user as protein sequence.</p>
</dd>
<dt><strong>dbCAN-sub.substrate.tsv</strong></dt><dd><p>Output from the pyHMMER using dbCAN_sub-HMM.</p>
</dd>
<dt><strong>diamond_results.tsv</strong></dt><dd><p>Results from the Diamond BLAST using CAZy.faa.</p>
</dd>
<dt><strong>dbCAN_hmm_results.tsv</strong></dt><dd><p>Output from the pyHMMER using dbCAN-HMM..</p>
</dd>
<dt><strong>overview.tsv</strong></dt><dd><p>Summarizes CAZyme predictions across tools. We recommend results using at least two tools (Shown as the “Recommend Results”).</p>
</dd>
</dl>
</section>
<section id="running-example-data-for-cgc-annotation-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues">
<h2>3. Running Example Data for CGC Annotation (please check the previous step for downloading example fasta data, we don’t repeat it here to avoid issues.)<a class="headerlink" href="#running-example-data-for-cgc-annotation-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues" title="Permalink to this heading"></a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.fna<span class="w"> </span>--mode<span class="w"> </span>prok<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_fna_CGC<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_gff<span class="w"> </span>gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_faa_CGC<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>EscheriaColiK12MG1655.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>NCBI_prok
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylona_heveae_TC161.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylona_heveae_TC161_faa_CGC<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>Xylona_heveae_TC161.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>NCBI_euk
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.aa.fasta<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylhe1_faa_CGC<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>JGI<span class="w"> </span>--input_gff<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>JGI
</pre></div>
</div>
</section>
<section id="id1">
<h2>4. Understanding the Output<a class="headerlink" href="#id1" title="Permalink to this heading"></a></h2>
<p>including the output files from the previous step, and new outputs:</p>
<dl class="simple">
<dt><strong>non_CAZyme.faa</strong></dt><dd><p>The non-CAZyme protein sequences extracted from uniInput.faa, which is based on the overview results.</p>
</dd>
<dt><strong>TC_results.tsv</strong></dt><dd><p>Results from the Diamond BLAST using TCDB to annotate transporter protein.</p>
</dd>
<dt><strong>TF_results.tsv</strong></dt><dd><p>Results from the pyHMMER using TF-HMM to annotate transcription factor protein.</p>
</dd>
<dt><strong>STP_results.tsv</strong></dt><dd><p>Results from the pyHMMER using STP-HMM to annotate signal transduction protein.</p>
</dd>
<dt><strong>total_cgc_info.tsv</strong></dt><dd><p>The total annotation of all signature proteins combing TC, TF, and STP. Using the same overlap method to filter as CAZyme annotation.</p>
</dd>
<dt><strong>cgc.gff</strong></dt><dd><p>The input file of CGCFinder in gff format. This is generated by the tool suite based on the input_gff file and “total_cgc_info.tsv”.</p>
</dd>
<dt><strong>cgc_standard_out.tsv</strong></dt><dd><p>The standard output of CGCFinder.</p>
</dd>
</dl>
</section>
<section id="running-example-data-for-substrate-prediction-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues-here-we-download-the-gff-files">
<h2>5. Running Example Data for Substrate Prediction (please check the previous step for downloading example fasta data, we don’t repeat it here to avoid issues. Here we download the gff files.)<a class="headerlink" href="#running-example-data-for-substrate-prediction-please-check-the-previous-step-for-downloading-example-fasta-data-we-don-t-repeat-it-here-to-avoid-issues-here-we-download-the-gff-files" title="Permalink to this heading"></a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.fna<span class="w"> </span>--mode<span class="w"> </span>prok<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_fna_sub<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_gff<span class="w"> </span>gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/NCBI_prok_test/EscheriaColiK12MG1655.gff<span class="w"> </span>-O<span class="w"> </span>EscheriaColiK12MG1655.gff
run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>EscheriaColiK12MG1655.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_EscheriaColiK12MG1655_faa_sub<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>EscheriaColiK12MG1655.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>NCBI_prok
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/NCBI_euk_test/Xylona_heveae_TC161.gff<span class="w"> </span>-O<span class="w"> </span>Xylona_heveae_TC161.gff
run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylona_heveae_TC161.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylona_heveae_TC161_faa_sub<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>Xylona_heveae_TC161.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>NCBI_euk
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/JGI_test/Xylhe1_GeneCatalog_proteins_20130827.gff<span class="w"> </span>-O<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.gff
run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.aa.fasta<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>output_Xylhe1_faa_sub<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>JGI<span class="w"> </span>--input_gff<span class="w"> </span>Xylhe1_GeneCatalog_proteins_20130827.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>JGI
</pre></div>
</div>
</section>
<section id="id2">
<h2>1. Understanding the Output<a class="headerlink" href="#id2" title="Permalink to this heading"></a></h2>
<p>including the output files from the previous step, and new outputs:</p>
<dl class="simple">
<dt><strong>substrate.out</strong></dt><dd><p>The final output of substrate prediction, which includes the substrate prediction results of each CAZyme gene cluster.</p>
</dd>
<dt><strong>PUL_blast.out</strong></dt><dd><p>The DIAMOND blastp results of CGCs against dbCAN-PULs.</p>
</dd>
<dt><strong>synteny_pdf/</strong></dt><dd><p>The synteny plot folder including predicted results. The plot shows the gene cluster mapping between PULs and CGCs.</p>
</dd>
</dl>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="prepare_the_database.html" class="btn btn-neutral float-left" title="Database Installation Command" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="input_fasta_data_processing.html" class="btn btn-neutral float-right" title="Process the input fasta file" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Xinpeng Zhang; Haidong Yi; Jinfang Zheng.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>