

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol &mdash; dbCAN-new-README 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=87f11076"></script>
      <script src="_static/doctools.js?v=888ff710"></script>
      <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" href="run_from_raw_reads_pr.html" />
    <link rel="prev" title="Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" href="run_from_raw_reads.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            dbCAN-new-README
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="prepare_the_database.html">Database Installation Command</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_fasta_data_processing.html">Process the input fasta file</a></li>
<li class="toctree-l1"><a class="reference internal" href="CAZyme_annotation.html">CAZyme Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_information_generation.html">Prepare the CGC annotation information</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_annotation.html">Run with CGCFinder</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_CGC_substrate.html">Substrate prediction with CGCs</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_substrate_plots.html">CGC substrate visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads.html">Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#example-2-amelia-2024-dataset-amelia-2024">Example 2: Amelia 2024 Dataset amelia_2024</a></li>
<li class="toctree-l2"><a class="reference internal" href="#procedure">Procedure</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#software-and-versions">Software and versions</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#database-installation-command">Database Installation Command</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-1-reads-processing-to-obtain-contigs">Module 1: Reads processing to obtain contigs</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p2-trim-adapter-and-low-quality-reads-timing-20min">P2. Trim adapter and low-quality reads (TIMING ~20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p3-assemble-reads-into-contigs-timing-84min">P3. Assemble reads into contigs (TIMING ~84min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p4-predict-genes-by-pyrodigal-timing-24min">P4. Predict genes by <cite>Pyrodigal</cite> (TIMING ~24min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-2-run-dbcan-annotation-fig-3-to-obtain-cazymes-cgcs-and-substrates">Module 2. run_dbcan annotation (Fig. 3) to obtain CAZymes, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min">P5. CAZyme annotation at the CAZyme family level (TIMING ~10min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p6-cgc-prediction-timing-15-min">P6. CGC prediction (TIMING ~15 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h">P7. Substrate prediction for CAZymes and CGCs (TIMING ~5h)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">Module 3. Read mapping (Fig. 3) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p8-read-mapping-to-all-contigs-of-each-sample-timing-10-min">P8. Read mapping to all contigs of each sample (TIMING ~10 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p9-sort-sam-files-by-coordinates-timing-6min">P9. Sort SAM files by coordinates (TIMING ~6min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-1min">P10. Read count calculation for all proteins of each sample using dbcan_utils (TIMING ~1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min">P11. Read count calculation for a given region of contigs using Samtools (TIMING ~1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">P12. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">Module 4: dbcan_plot for data visualization (Fig. 3) of abundances of CAZymes, CGCs, and substrates (TIMING variable)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p13-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min">P13. Heatmap for CAZyme substrate abundance across samples (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s4c-timing-1min">P14. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. S4C) (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-s4a-timing-1min">P15. Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC (Fig. S4A) (TIMING 1min)</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_pr.html">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_wk.html">Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_em.html">Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_co_assem.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_subsample.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_assem_free.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">dbCAN-new-README</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/run_from_raw_reads_am.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="run-from-raw-reads-amelia-2024-automated-cazyme-and-glycan-substrate-annotation-in-microbiomes-a-step-by-step-protocol">
<h1>Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol<a class="headerlink" href="#run-from-raw-reads-amelia-2024-automated-cazyme-and-glycan-substrate-annotation-in-microbiomes-a-step-by-step-protocol" title="Permalink to this heading"></a></h1>
<section id="example-2-amelia-2024-dataset-amelia-2024">
<h2>Example 2: Amelia 2024 Dataset <a class="reference external" href="https://academic.oup.com/ismecommun/article/4/1/ycae108/7742758#480953371">amelia_2024</a><a class="headerlink" href="#example-2-amelia-2024-dataset-amelia-2024" title="Permalink to this heading"></a></h2>
<p>The Amelia 2024 dataset <a class="reference external" href="https://academic.oup.com/ismecommun/article/4/1/ycae108/7742758#480953371">amelia_2024</a> was published in 2024 from a soil metagenomic study. In the published paper. They analyzed the effects of wildfires on the soil microbiome.</p>
<p>The raw read data, intermediate data from each analysis step, and final result data and visualization files are organized in nested folders available on our website <a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/">https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/</a>).</p>
</section>
<section id="procedure">
<h2>Procedure<a class="headerlink" href="#procedure" title="Permalink to this heading"></a></h2>
<section id="software-and-versions">
<h3>Software and versions<a class="headerlink" href="#software-and-versions" title="Permalink to this heading"></a></h3>
<ul class="simple">
<li><p><strong>Anaconda</strong> (<a class="reference external" href="https://www.anaconda.com">Anaconda</a>, version 23.7.3)</p></li>
<li><p><strong>MEGAHIT</strong> (<a class="reference external" href="https://github.com/voutcn/megahit">MEGAHIT</a>, version 1.2.9)</p></li>
<li><p><strong>pyHMMER</strong> (<a class="reference external" href="https://pyhmmer.readthedocs.io/en/stable/">pyHMMER</a>, newest version)</p></li>
<li><p><strong>DIAMOND</strong> (<a class="reference external" href="https://github.com/bbuchfink/diamond">DIAMOND</a>, version 2.1.8)</p></li>
<li><p><strong>TrimGalore</strong> (<a class="reference external" href="https://github.com/FelixKrueger/TrimGalore">TrimGalore</a>, version 0.6.0)</p></li>
<li><p><strong>Pyrodigal</strong> (<a class="reference external" href="https://pyrodigal.readthedocs.io/en/stable/">Pyrodigal</a>, newest version)</p></li>
<li><p><strong>BWA</strong> (<a class="reference external" href="https://github.com/lh3/bwa">BWA</a>, version 0.7.17)</p></li>
<li><p><strong>Samtools</strong> (<a class="reference external" href="https://github.com/samtools/samtools">Samtools</a>, version 1.7)</p></li>
<li><p><strong>Kraken2</strong> (<a class="reference external" href="https://ccb.jhu.edu/software/kraken2/">Kraken2</a>, version 2.1.1)</p></li>
<li><p><strong>KrakenTools</strong> (<a class="reference external" href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file">KrakenTools</a>, version 1.2)</p></li>
<li><p><strong>run_dbcan</strong> (<a class="reference external" href="https://github.com/Xinpeng021001/run_dbCAN_new">run_dbcan</a>, version 5.0.3)</p></li>
<li><p><strong>MMseqs2</strong> (<a class="reference external" href="https://github.com/soedinglab/MMseqs2">MMseqs2</a>, release 15-6f452)</p></li>
<li><p><strong>Pysam</strong> (<a class="reference external" href="https://github.com/pysam-developers/pysam">Pysam</a>, version 0.22.1)</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>CAZyme_annotation_am.yml
conda<span class="w"> </span>activate<span class="w"> </span>CAZyme_annotation_am
</pre></div>
</div>
<p>To install the databases, execute the following commands:</p>
<section id="database-installation-command">
<h4>Database Installation Command<a class="headerlink" href="#database-installation-command" title="Permalink to this heading"></a></h4>
<p>prepare the conda environment (available at <a class="reference external" href="https://github.com/bcb-unl/run_dbcan_new/tree/master/envs">https://github.com/bcb-unl/run_dbcan_new/tree/master/envs</a>):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yml
conda<span class="w"> </span>activate<span class="w"> </span>run_dbcan_env
</pre></div>
</div>
<ol class="arabic simple">
<li><p>We provide command in the run_dbcan script:</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>database<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Users could also download all  database files from the dbCAN2 website (<a class="reference external" href="http://bcb.unl.edu/dbCAN2/download/Databases/">http://bcb.unl.edu/dbCAN2/download/Databases/</a>), and then put them into the db directory.</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/run_dbCAN_database_total/dbCAN_db.tar.gz<span class="w"> </span>-O<span class="w"> </span>db.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>db.tar.gz
rm<span class="w"> </span>db.tar.gz
</pre></div>
</div>
</section>
</section>
<section id="module-1-reads-processing-to-obtain-contigs">
<h3>Module 1: Reads processing to obtain contigs<a class="headerlink" href="#module-1-reads-processing-to-obtain-contigs" title="Permalink to this heading"></a></h3>
<p>S1| Download Amelia2024 (Table 2) raw reads (~20min)
.. code-block:: shell</p>
<blockquote>
<div><p>wget <a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HTOOS_1.fastq.gz">https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HTOOS_1.fastq.gz</a>
wget <a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HTOOS_2.fastq.gz">https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HTOOS_2.fastq.gz</a>
wget <a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU_1.fastq.gz">https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU_1.fastq.gz</a>
wget <a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU_2.fastq.gz">https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU_2.fastq.gz</a></p>
</div></blockquote>
<p>These raw data were originally downloaded from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR24887509">https://www.ncbi.nlm.nih.gov/sra/?term=SRR24887509</a> and <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR24887498">https://www.ncbi.nlm.nih.gov/sra/?term=SRR24887498</a> and renamed to indicate their collected samples (Table 2).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2-build<span class="w"> </span>--standard<span class="w"> </span>--db<span class="w"> </span>K2
</pre></div>
</div>
<p>or could download database directly from the <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">https://benlangmead.github.io/aws-indexes/k2</a>,
and extract the tarball to the db folder, here we use standard database as an example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://genome-idx.s3.amazonaws.com/kraken/k2_standard_20240605.tar.gz
mkdir<span class="w"> </span>K2
tar<span class="w"> </span>-xvf<span class="w"> </span>k2_standard_20240605.tar.gz<span class="w"> </span>-C<span class="w"> </span>K2
</pre></div>
</div>
<p>Use <cite>kraken2</cite> to check for contaminated reads:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>HTOOS.kreport<span class="w"> </span>--output<span class="w"> </span>HTOOS.kraken.output<span class="w"> </span>HTOOS_1.fastq.gz<span class="w"> </span>HTOOS_2.fastq.gz

kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>HCZGU.kreport<span class="w"> </span>--output<span class="w"> </span>HCZGU.kraken.output<span class="w"> </span>HCZGU_1.fastq.gz<span class="w"> </span>HCZGU_2.fastq.gz
</pre></div>
</div>
<p>Kraken2 found very little contamination in the Amelia2024 data. Consequently, there was no need for the contamination removal step.</p>
<p>If contamination is identified, users can align the reads to the reference genomes of potential contamination source organisms to remove
the aligned reads (Box 1). The most common source in human microbiome studies is from human hosts.</p>
<blockquote>
<div><p>Kraken2 will produce the following output files:</p>
</div></blockquote>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">9</span>.2G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">05</span>:59<span class="w"> </span>HTOOS.kraken.output
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">1</span>.2M<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">05</span>:59<span class="w"> </span>HTOOS.kreport
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">4</span>.9G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:21<span class="w"> </span>HCZGU.kraken.output
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">1</span>.1M<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:21<span class="w"> </span>HCZGU.kreport
</pre></div>
</div>
<p>Suppose from these files, we have identified humans as the contam</p>
<p>Suppose from these files, we have identified humans as the contamination source, we can use the following commands to remove the contamination reads by aligning reads to the human reference genome.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>index<span class="w"> </span>-p<span class="w"> </span>hg38<span class="w"> </span>Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>HTOOS_1.fastq.gz<span class="w"> </span>HTOOS_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>HTOOS.hg38.sam
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>HCZGU_1.fastq.gz<span class="w"> </span>HCZGU_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>HCZGU.hg38.sam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>HCZGU.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>HCZGU.hg38.unmap.bam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>HTOOS.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>HTOOS.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>HCZGU_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>HCZGU_2.clean.fq.gz<span class="w"> </span>HCZGU.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>HTOOS_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>HTOOS_2.clean.fq.gz<span class="w"> </span>HTOOS.hg38.unmap.bam
</pre></div>
</div>
<p>KrakenTools could also extract host reads quickly and easied which is recommended. We use tax 2759 (plant) as an example.
Please read KrakenTools README for more information (<a class="reference external" href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file">https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file</a>).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>HTOOS.kraken.output<span class="w">  </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s1<span class="w"> </span>HTOOS_1.fq.gz<span class="w"> </span>-s2<span class="w"> </span>HTOOS_2.fastq.gz<span class="w"> </span>-o<span class="w"> </span>HTOOS_1.clean.fq<span class="w"> </span>-o2<span class="w"> </span>HTOOS_2.clean.fq
gzip<span class="w"> </span>HTOOS_1.clean.fq
gzip<span class="w"> </span>HTOOS_2.clean.fq


extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>HCZGU.kraken.output<span class="w"> </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s1<span class="w"> </span>HCZGU_1.fastq.gz<span class="w"> </span>-s2<span class="w"> </span>HCZGU_2.fastq.gz<span class="w"> </span>-o<span class="w"> </span>HCZGU_1.clean.fq<span class="w"> </span>-o2<span class="w"> </span>HCZGU_2.clean.fq
gzip<span class="w"> </span>HCZGU_1.clean.fq
gzip<span class="w"> </span>HCZGU_2.clean.fq
</pre></div>
</div>
<section id="p2-trim-adapter-and-low-quality-reads-timing-20min">
<h4>P2. Trim adapter and low-quality reads (TIMING ~20min)<a class="headerlink" href="#p2-trim-adapter-and-low-quality-reads-timing-20min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>trim_galore<span class="w"> </span>--paired<span class="w"> </span>HTOOS_1.fastq.gz<span class="w"> </span>HTOOS_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>

trim_galore<span class="w"> </span>--paired<span class="w"> </span>HCZGU_1.fastq.gz<span class="w"> </span>HCZGU_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<p>We specified –illumina to indicate that the reads were generated using the Illumina sequencing platform.
Nonetheless, trim_galore can automatically detect adapters, providing flexibility for users who may know the specific sequencing platform.
Details of trimming are available in the trimming report file (Box 2).</p>
<section id="box-2-example-output-of-trim-galore">
<h5>Box 2: Example output of <cite>trim_galore</cite><a class="headerlink" href="#box-2-example-output-of-trim-galore" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In addition to the trimmed read files, <cite>Trim_galore`</cite> also generates a trimming report file.
The trimming report contains details on read trimming, such as the number of trimmed reads.</p>
</div></blockquote>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">5</span>.1K<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:27<span class="w"> </span>HCZGU_1.fastq.gz_trimming_report.txt
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">8</span>.5G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:45<span class="w"> </span>HCZGU_1_val_1.fq.gz
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">5</span>.3K<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:45<span class="w"> </span>HCZGU_2.fastq.gz_trimming_report.txt
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">8</span>.5G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">03</span>:45<span class="w"> </span>HCZGU_2_val_2.fq.gz
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">5</span>.2K<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">06</span>:10<span class="w"> </span>HTOOS_1.fastq.gz_trimming_report.txt
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">  </span>16G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">06</span>:45<span class="w"> </span>HTOOS_1_val_1.fq.gz
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">5</span>.4K<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">06</span>:45<span class="w"> </span>HTOOS_2.fastq.gz_trimming_report.txt
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">  </span>16G<span class="w"> </span>May<span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">06</span>:45<span class="w"> </span>HTOOS_2_val_2.fq.gz
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>During the trimming process, certain reads may be entirely removed due to low quality in its entirety.
Using the <code class="docutils literal notranslate"><span class="pre">--retain_unpaired</span></code> parameter in <code class="docutils literal notranslate"><span class="pre">trim_galore</span></code> allows for the preservation of single-end reads.
In this protocol, this option was not selected, so that both reads of a forward-revise pair were removed.</p>
</div>
</section>
</section>
<section id="p3-assemble-reads-into-contigs-timing-84min">
<h4>P3. Assemble reads into contigs (TIMING ~84min)<a class="headerlink" href="#p3-assemble-reads-into-contigs-timing-84min" title="Permalink to this heading"></a></h4>
<p>Use Megahit for assembling reads into contigs:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>megahit<span class="w"> </span>-m<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>megahit_HTOOS<span class="w"> </span>-1<span class="w"> </span>HTOOS_1_val_1.fq.gz<span class="w"> </span>-2<span class="w"> </span>HTOOS_2_val_2.fq.gz<span class="w"> </span>--out-prefix<span class="w"> </span>HTOOS<span class="w"> </span>--min-contig-len<span class="w"> </span><span class="m">1000</span>

megahit<span class="w"> </span>-m<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>megahit_HCZGU<span class="w"> </span>-1<span class="w"> </span>HCZGU_1_val_1.fq.gz<span class="w"> </span>-2<span class="w"> </span>HCZGU_2_val_2.fq.gz<span class="w"> </span>--out-prefix<span class="w"> </span>HCZGU<span class="w"> </span>--min-contig-len<span class="w"> </span><span class="m">1000</span>
</pre></div>
</div>
<p>MEGAHIT generates two output folders: megahit_HCZGU and megahit_HTOOS.
Each contains five files and one sub-folder (Box 3). HTOOS.contigs.fa is the final contig sequence file. We set –min-contig-len 1000, a common practice to retain all contigs longer than 1,000 base pairs.</p>
<section id="box-3-example-output-of-megahit">
<h5>Box 3: Example output of <cite>MEGAHIT</cite><a class="headerlink" href="#box-3-example-output-of-megahit" title="Permalink to this heading"></a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">   </span><span class="m">262</span><span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">04</span>:58<span class="w"> </span>checkpoints.txt
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">     </span><span class="m">0</span><span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">04</span>:58<span class="w"> </span><span class="k">done</span>
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">  </span><span class="m">2</span>.1G<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">04</span>:58<span class="w"> </span>HTOOS.contigs.fa
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">  </span><span class="m">1</span>.4M<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">04</span>:58<span class="w"> </span>HTOOS.log
drwxr-xr-x<span class="w"> </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">   </span>41K<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">04</span>:57<span class="w"> </span>intermediate_contigs
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w">  </span><span class="m">1</span>.1K<span class="w"> </span>Jun<span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="m">18</span>:59<span class="w"> </span>options.json
</pre></div>
</div>
</section>
</section>
<section id="p4-predict-genes-by-pyrodigal-timing-24min">
<h4>P4. Predict genes by <cite>Pyrodigal</cite> (TIMING ~24min)<a class="headerlink" href="#p4-predict-genes-by-pyrodigal-timing-24min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pyrodigal<span class="w"> </span>-p<span class="w"> </span>meta<span class="w"> </span>-i<span class="w"> </span>megahit_HTOOS/HTOOS.contigs.fa<span class="w"> </span>-d<span class="w"> </span>fefifo_8022_1.cds<span class="w"> </span>-a<span class="w"> </span>fefifo_8022_1.faa<span class="w"> </span>-f<span class="w"> </span>gff<span class="w"> </span>-o<span class="w"> </span>fefifo_8022_1.gff<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
pyrodigal<span class="w"> </span>-p<span class="w"> </span>meta<span class="w"> </span>-i<span class="w"> </span>megahit_HCZGU/HCZGU.contigs.fa<span class="w"> </span>-d<span class="w"> </span>fefifo_8022_7.cds<span class="w"> </span>-a<span class="w"> </span>fefifo_8022_7.faa<span class="w"> </span>-f<span class="w"> </span>gff<span class="w"> </span>-o<span class="w"> </span>fefifo_8022_7.gff<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<section id="box-4-example-output-of-pyrodigal">
<h5>Box 4: Example output of <cite>Pyrodigal</cite><a class="headerlink" href="#box-4-example-output-of-pyrodigal" title="Permalink to this heading"></a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span><span class="m">2</span>.1G<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">07</span>:15<span class="w"> </span>HTOOS.cds
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span>921M<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">07</span>:15<span class="w"> </span>HTOOS.faa
-rw-r--r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>yinlab<span class="w"> </span>857M<span class="w"> </span>Jun<span class="w">  </span><span class="m">3</span><span class="w"> </span><span class="m">07</span>:15<span class="w"> </span>HTOOS.gff
</pre></div>
</div>
<p>Hint: If you apply <code class="docutils literal notranslate"><span class="pre">prodigal</span></code> not <code class="docutils literal notranslate"><span class="pre">pyrodigal</span></code>, use this to fix gene ID in gff files by dbcan_utils (TIMING ~1min)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_utils<span class="w"> </span>gff_fix<span class="w"> </span>-i<span class="w"> </span>HTOOS.faa<span class="w"> </span>-g<span class="w"> </span>HTOOS.gff
dbcan_utils<span class="w"> </span>gff_fix<span class="w"> </span>-i<span class="w"> </span>HCZGU.faa<span class="w"> </span>-g<span class="w"> </span>HCZGU.gff
</pre></div>
</div>
</section>
</section>
</section>
<section id="module-2-run-dbcan-annotation-fig-3-to-obtain-cazymes-cgcs-and-substrates">
<h3>Module 2. run_dbcan annotation (Fig. 3) to obtain CAZymes, CGCs, and substrates<a class="headerlink" href="#module-2-run-dbcan-annotation-fig-3-to-obtain-cazymes-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<p><strong>CRITICAL STEP</strong></p>
<p>Users can skip P5 and P6, and directly run P7 (much slower though), if they want to predict not only CAZymes and CGCs, but also substrates.</p>
<section id="p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min">
<h4>P5. CAZyme annotation at the CAZyme family level (TIMING ~10min)<a class="headerlink" href="#p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>HTOOS.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>HTOOS.CAZyme<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI
run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>HCZGU.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>HCZGU.CAZyme<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI
</pre></div>
</div>
<p>Two arguments are required for <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code>: the input sequence file (faa files) and the sequence type (protein).
By default, <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> will use three methods (<code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN</span> <span class="pre">HMMdb</span></code>, <code class="docutils literal notranslate"><span class="pre">DIAMOND</span></code> vs <code class="docutils literal notranslate"><span class="pre">CAZy</span></code>, <code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN-sub</span> <span class="pre">HMMdb</span></code>) for
CAZyme annotation.</p>
<section id="box-5-cazyme-annotation-with-default-setting">
<h5>Box 5: CAZyme annotation with default setting<a class="headerlink" href="#box-5-cazyme-annotation-with-default-setting" title="Permalink to this heading"></a></h5>
<p>The sequence type can be <cite>protein</cite>, <cite>prok</cite>, <cite>meta</cite>. If the input sequence file contains metagenomic contig sequences (<cite>fna</cite> file),
the sequence type has to be <cite>meta</cite>, and <cite>prodigal</cite> will be called to predict genes.</p>
</section>
</section>
<section id="p6-cgc-prediction-timing-15-min">
<h4>P6. CGC prediction (TIMING ~15 min)<a class="headerlink" href="#p6-cgc-prediction-timing-15-min" title="Permalink to this heading"></a></h4>
<p>The following commands will re-run run_dbcan to not only predict CAZymes but also CGCs with protein <cite>faa</cite> and gene location <cite>gff</cite> files.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>HTOOS.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>HTOOS.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>HTOOS.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>HCZGU.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>HCZGU.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>HCZGU.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p><strong>Creating own gff file</strong>
If the users would like to create their own <code class="docutils literal notranslate"><span class="pre">gff</span></code> file (instead of using Prokka or Prodigal),
it is important to make sure the value of ID attribute in the <code class="docutils literal notranslate"><span class="pre">gff</span></code> file matches the protein ID in the protein <code class="docutils literal notranslate"><span class="pre">faa</span></code> file.</p>
<p><strong>[Troubleshooting]CGC not found</strong>
If no result is found in CGC output file, it is most likely because the sequence IDs in <code class="docutils literal notranslate"><span class="pre">gff</span></code> file and <code class="docutils literal notranslate"><span class="pre">faa</span></code> file do not match.
Another less likely reason is that the contigs are too short and fragmented and not suitable for CGC prediction.</p>
</div>
</section>
<section id="p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h">
<h4>P7. Substrate prediction for CAZymes and CGCs (TIMING ~5h)<a class="headerlink" href="#p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h" title="Permalink to this heading"></a></h4>
<p>The following commands will re-run run_dbcan to predict CAZymes, CGCs, and their substrates.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>HTOOS.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>HTOOS.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w">   </span>--output_dir<span class="w"> </span>HTOOS.dbCAN<span class="w"> </span>--db_dir<span class="w"> </span>db
run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>HCZGU.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>HCZGU.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w">   </span>--output_dir<span class="w"> </span>HCZGU.dbCAN<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>In the <cite>HCZGU.dbCAN &lt;https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU.dbCAN/&gt;_</cite> directory, a total of 17 files and 1 folder are generated.</p>
</section>
</section>
<section id="module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">
<h3>Module 3. Read mapping (Fig. 3) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates<a class="headerlink" href="#module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<section id="p8-read-mapping-to-all-contigs-of-each-sample-timing-10-min">
<h4>P8. Read mapping to all contigs of each sample (TIMING ~10 min)<a class="headerlink" href="#p8-read-mapping-to-all-contigs-of-each-sample-timing-10-min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>samfiles
bwa<span class="w"> </span>index<span class="w"> </span>megahit_HCZGU/HCZGU.contigs.fa
bwa<span class="w"> </span>index<span class="w"> </span>megahit_HTOOS/HTOOS.contigs.fa
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/HCZGU.sam<span class="w"> </span>megahit_HCZGU/HCZGU.contigs.fa<span class="w"> </span>HCZGU_1_val_1.fq.gz<span class="w"> </span>HCZGU_2_val_2.fq.gz
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/HTOOS.sam<span class="w"> </span>megahit_HTOOS/HTOOS.contigs.fa<span class="w"> </span>HTOOS_1_val_1.fq.gz<span class="w"> </span>HTOOS_2_val_2.fq.gz
</pre></div>
</div>
</section>
<section id="p9-sort-sam-files-by-coordinates-timing-6min">
<h4>P9. Sort SAM files by coordinates (TIMING ~6min)<a class="headerlink" href="#p9-sort-sam-files-by-coordinates-timing-6min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>samfiles
samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>HCZGU.bam<span class="w"> </span>HCZGU.sam
samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>HTOOS.bam<span class="w"> </span>HTOOS.sam
samtools<span class="w"> </span>index<span class="w"> </span>HCZGU.bam
samtools<span class="w"> </span>index<span class="w"> </span>HTOOS.bam
rm<span class="w"> </span>-rf<span class="w"> </span>*sam
<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
</section>
<section id="p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-1min">
<h4>P10. Read count calculation for all proteins of each sample using dbcan_utils (TIMING ~1min)<a class="headerlink" href="#p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>HCZGU_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>HCZGU_abund
dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../HCZGU.gff<span class="w"> </span>-i<span class="w">  </span>../samfiles/HCZGU.bam<span class="w"> </span>-o<span class="w"> </span>HCZGU.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>--identity<span class="w"> </span><span class="m">0</span>.98

<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>mkdir<span class="w"> </span>HTOOS_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>HTOOS_abund
dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../HTOOS.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/HTOOS.bam<span class="w"> </span>-o<span class="w"> </span>HTOOS.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>--identity<span class="w"> </span><span class="m">0</span>.98
<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
</section>
<section id="p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min">
<h4>P11. Read count calculation for a given region of contigs using Samtools (TIMING ~1min)<a class="headerlink" href="#p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>HCZGU_abund
samtools<span class="w"> </span>depth<span class="w"> </span>-r<span class="w"> </span>k141_1169008:2-5684<span class="w"> </span>../samfiles/HCZGU.bam<span class="w"> </span>&gt;<span class="w"> </span>HCZGU.cgc.depth.txt
</pre></div>
</div>
<p>The parameter -r k141_1169008:2-5684 specifies a region in a contig. For any CGC, its positional range can be found in the file</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The contig IDs are automatically generated by MEGAHIT. There is a small chance that a same contig ID appears in both samples. However, the two contigs in the two samples do not match each other even the ID is the same. For example, the contig ID <code class="docutils literal notranslate"><span class="pre">k141_2168</span></code> is most likely only found in the <code class="docutils literal notranslate"><span class="pre">fefifo_8022_1</span></code> sample. Even if there is a <code class="docutils literal notranslate"><span class="pre">k141_2168</span></code> in <code class="docutils literal notranslate"><span class="pre">fefifo_8022_7</span></code>, the actual contigs in two samples are different.</p>
</div>
</section>
<section id="p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">
<h4>P12. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)<a class="headerlink" href="#p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>HCZGU.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HCZGU.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>HCZGU.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HCZGU.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>HCZGU.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HCZGU.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>HCZGU.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HCZGU.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>HTOOS_abund
dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>HTOOS.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HTOOS.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>HTOOS.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HTOOS.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>HTOOS.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HTOOS.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>HTOOS.depth.txt<span class="w"> </span>-i<span class="w"> </span>../HTOOS.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM
<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>We developed a set of Python scripts as <code class="docutils literal notranslate"><span class="pre">dbcan_utils</span></code> (included in the <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> package) to take the raw read counts for all genes as input and output the normalized abundances (refer to Box 7) of CAZyme families, subfamilies, CGCs, and substrates (see Fig. 4). The parameter <code class="docutils literal notranslate"><span class="pre">-a</span> <span class="pre">TPM</span></code> can also be set to two other metrics: RPM, or RPKM61.</p>
<ul class="simple">
<li><p><strong>RPKM</strong> is calculated as the number of mapped reads to a gene G divided by [(total number of mapped reads to all genes / 10^6) x (gene G length / 1000)].</p></li>
<li><p><strong>RPM</strong> is the number of mapped reads to a gene G divided by (total number of mapped reads to all genes / 10^6).</p></li>
<li><p><strong>TPM</strong> is calculated as [number of mapped reads to a gene G / (gene G length / 1000)] divided by the sum of [number of mapped reads to each gene / (the gene length / 1000)].</p></li>
</ul>
<section id="box-7-example-output-of-dbcan-utils">
<h5>Box 7. Example output of dbcan_utils<a class="headerlink" href="#box-7-example-output-of-dbcan-utils" title="Permalink to this heading"></a></h5>
<p>As an example, <cite>HCZGU_abund  &lt;https://bcb.unl.edu/dbCAN_tutorial/dataset5-Amelia2024/HCZGU_abund/&gt;_</cite> folder has 7 TSV files:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>178K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>CGC_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.3K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>CGC_substrate_majority_voting.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>12K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>CGC_substrate_PUL_homology.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.5K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>EC_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.1K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>fam_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>42K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>fam_substrate_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>26K<span class="w"> </span>Jan<span class="w">  </span><span class="m">2</span><span class="w"> </span><span class="m">04</span>:08<span class="w"> </span>subfam_abund.out
</pre></div>
</div>
<p>Explanation of columns in these TSV files is as follows:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">fam_abund.out</span></code>: CAZy family (from HMMER vs dbCAN HMMdb), sum of TPM, number of CAZymes in the family.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">subfam_abund.out</span></code>: eCAMI subfamily (from HMMER vs dbCAN-sub HMMdb), sum of TPM, number of CAZymes in the subfamily.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">EC_abund.out</span></code>: EC number (extracted from dbCAN-sub subfamily), sum of TPM, number of CAZymes with the EC.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fam_substrate_abund.out</span></code>: Substrate (from HMMER vs dbCAN-sub HMMdb), sum of TPM (all CAZymes in this substrate group), GeneID (all CAZyme IDs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_abund.out</span></code>: CGC_ID (e.g., k141_338400|CGC1), mean of TPM (all genes in the CGC), Seq_IDs (IDs of all genes in the CGC), TPM (of all genes in the CGC), Families (CAZyme family or other signature gene type of all genes in the CGC).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_PUL_homology.out</span></code>: Substrate (from dbCAN-PUL blast search), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-PUL blast search), TPM (of CGCs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_majority_voting.out</span></code>: Substrate (from dbCAN-sub majority voting), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-sub majority voting), TPM (of CGCs in this substrate group).</p></li>
</ul>
</div></blockquote>
</section>
</section>
</section>
<section id="module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">
<h3>Module 4: dbcan_plot for data visualization (Fig. 3) of abundances of CAZymes, CGCs, and substrates (TIMING variable)<a class="headerlink" href="#module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable" title="Permalink to this heading"></a></h3>
<p><strong>CRITICAL STEP</strong></p>
<p>To visualize the CAZyme annotation result, we provide a set of Python scripts as dbcan_plot to make publication quality plots with the dbcan_utils results as the input. The dbcan_plot scripts are included in the run_dbcan package. Once the plots are made in PDF format, they can be transferred to users’ Windows or Mac computers for visualization.</p>
<p>Five data folders will be needed as the input for <code class="docutils literal notranslate"><span class="pre">dbcan_plot</span></code>:</p>
<ol class="arabic simple">
<li><p>two abundance folders <code class="docutils literal notranslate"><span class="pre">HCZGU_abund</span></code> and <code class="docutils literal notranslate"><span class="pre">HTOOS_abund</span></code>,</p></li>
<li><p>two CAZyme annotation <code class="docutils literal notranslate"><span class="pre">HCZGU.dbCAN</span></code> and <code class="docutils literal notranslate"><span class="pre">HTOOS.dbCAN</span></code>, and</p></li>
<li><p>the <code class="docutils literal notranslate"><span class="pre">dbCAN-PUL</span> <span class="pre">folder</span></code> (under the db folder, released from <code class="docutils literal notranslate"><span class="pre">dbCAN-PUL.tar.gz</span></code>).</p></li>
</ol>
<section id="p13-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min">
<h4>P13. Heatmap for CAZyme substrate abundance across samples (TIMING 1min)<a class="headerlink" href="#p13-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>heatmap_plot<span class="w"> </span>--samples<span class="w"> </span>HCZGU,HTOOS<span class="w"> </span>-i<span class="w"> </span>HCZGU_abund/fam_substrate_abund.out,HTOOS_abund/fam_substrate_abund.out<span class="w"> </span>--show_abund<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span>
</pre></div>
</div>
<p>Here we plot the top 20 substrates in the two samples. The input files are the two CAZyme substrate abundance files calculated based on dbCAN-sub result. The default heatmap is ranked by substrate abundances. To rank the heatmap according to abundance profile using the function clustermap of seaborn package, users can invoke the <code class="docutils literal notranslate"><span class="pre">--cluster_map</span></code> parameter.</p>
</section>
<section id="p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s4c-timing-1min">
<h4>P14. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. S4C) (TIMING 1min)<a class="headerlink" href="#p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s4c-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>HCZGU,HTOOS<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>HCZGU_abund/fam_abund.out,HTOOS_abund/fam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>fam.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>HCZGU,HTOOS<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>HCZGU_abund/subfam_abund.out,HTOOS_abund/subfam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>subfam.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>HCZGU,HTOOS<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>HCZGU_abund/EC_abund.out,HTOOS_abund/EC_abund.out<span class="w"> </span>--pdf<span class="w"> </span>ec.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>Users can choose to generate a barplot instead of heatmap using the <code class="docutils literal notranslate"><span class="pre">bar_plot</span></code> method.</p>
</section>
<section id="p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-s4a-timing-1min">
<h4>P15. Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC (Fig. S4A) (TIMING 1min)<a class="headerlink" href="#p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-s4a-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_coverage_plot<span class="w"> </span>-i<span class="w"> </span>HCZGU.dbCAN<span class="w"> </span>--readscount<span class="w"> </span>HCZGU_abund/HCZGU.cgc.depth.txt<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;‘k141_1169008|CGC1&#39;</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">HCZGU.dbCAN</span></code> folder contains the <code class="docutils literal notranslate"><span class="pre">PUL_blast.out</span></code> file. Using this file, the <code class="docutils literal notranslate"><span class="pre">cgc_standard.out</span></code> file,
and the best PUL’s <code class="docutils literal notranslate"><span class="pre">gff</span></code> file in <code class="docutils literal notranslate"><span class="pre">dbCAN-PUL.tar.gz</span></code>, the CGC_synteny_plot method will create the <code class="docutils literal notranslate"><span class="pre">CGC-PUL</span> <span class="pre">synteny</span> <span class="pre">plot</span></code>.
The <code class="docutils literal notranslate"><span class="pre">-cgcid</span></code> parameter is required to specify which CGC to be plotted (<code class="docutils literal notranslate"><span class="pre">'k141_1169008|CGC1'</span></code> in this example).
The <code class="docutils literal notranslate"><span class="pre">HCZGU.cgc.depth.txt</span></code> file is used to plot the read mapping coverage.</p>
<p>If users only want to plot the CGC structure:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_plot<span class="w"> </span>-i<span class="w"> </span>HCZGU.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_1169008|CGC1&#39;</span>
</pre></div>
</div>
<p>If users only want to plot the CGC structure plus the read mapping coverage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_coverage_plot<span class="w"> </span>-i<span class="w"> </span>HCZGU.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_1169008|CGC1&#39;</span><span class="w">  </span>--readscount<span class="w"> </span>HCZGU_abund/HCZGU.cgc.depth.txt
</pre></div>
</div>
<p>If users only want to plot the synteny between the CGC and PUL:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_plot<span class="w"> </span>-i<span class="w"> </span>HCZGU.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_1169008|CGC1&#39;</span>
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The CGC IDs in different samples do not match each other. For example, specifying -i HCZGU.dbCAN is to plot the ‘k141_1169008|CGC1’ in the HCZGU sample.
The ‘k141_1169008|CGC1’ in the HTOOS sample most likely does not exist, and even it does, the CGC has a different sequence even if the ID is the same.</p>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="run_from_raw_reads.html" class="btn btn-neutral float-left" title="Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="run_from_raw_reads_pr.html" class="btn btn-neutral float-right" title="Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Xinpeng Zhang; Haidong Yi; Jinfang Zheng.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>