

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol &mdash; dbCAN-new-README 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=87f11076"></script>
      <script src="_static/doctools.js?v=888ff710"></script>
      <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" href="run_from_raw_reads_wk.html" />
    <link rel="prev" title="Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" href="run_from_raw_reads_am.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            dbCAN-new-README
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="prepare_the_database.html">Database Installation Command</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_fasta_data_processing.html">Process the input fasta file</a></li>
<li class="toctree-l1"><a class="reference internal" href="CAZyme_annotation.html">CAZyme Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_information_generation.html">Prepare the CGC annotation information</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_annotation.html">Run with CGCFinder</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_CGC_substrate.html">Substrate prediction with CGCs</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_substrate_plots.html">CGC substrate visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads.html">Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_am.html">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#example-3-priest2023-dataset-priest-2023">Example 3: Priest2023 Dataset priest_2023</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#software-and-versions">Software and versions</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#database-installation-command">Database Installation Command</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-1-reads-processing-fig-3-to-obtain-contigs">Module 1: Reads processing (Fig. 3) to obtain contigs</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p1-contamination-check-timing-10min">P1. Contamination Check (TIMING ~10min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p2-remove-contamination-reads-from-human-timing-40min">P2. Remove contamination reads from human (TIMING ~40min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p3-trim-adapter-and-low-quality-reads-timing-20min">P3| Trim adapter and low-quality reads (TIMING ~20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p4-assemble-hifi-reads-into-metagenome-timing-4h20min">P4. Assemble HiFi reads into metagenome (TIMING ~4h20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p5-predict-genes-by-prodigal-14min">P5. Predict genes by Prodigal (~14min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-1-run-dbcan-annotation-to-obtain-cazymes-cgcs-and-substrates">Module 1: run_dbcan annotation to obtain CAZymes, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p6-cazyme-annotation-at-family-level-timing-10min">P6. CAZyme annotation at family level (TIMING ~10min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p7-cgc-prediction-timing-15-min">P7. CGC prediction (TIMING ~15 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p8-substrate-prediction-for-cazymes-and-cgcs-timing-5h">P8. Substrate prediction for CAZymes and CGCs (TIMING ~5h)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">Module 3. Read mapping (Fig. 3) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p9-read-mapping-to-all-contigs-of-each-sample-timing-10-min">P9. Read mapping to all contigs of each sample (TIMING ~10 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p10-hifi-mg-long-read-mapping-to-all-contigs-of-each-sample-timing-20min">P10. HiFi MG long read mapping to all contigs of each sample (TIMING ~20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p12-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utilsbedtools-timing-6min12min">P12. Read count calculation for all proteins of each sample using dbcan_utilsBedtools (TIMING ~6min12min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p13-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min">P13. Read count calculation for a given region of contigs using Samtools (TIMING ~1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p14-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">P14. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">Module 4: dbcan_plot for data visualization (Fig. 3) of abundances of CAZymes, CGCs, and substrates (TIMING variable)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p15-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min">P15. Heatmap for CAZyme substrate abundance across samples (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p16-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s5a-c-timing-1min">P16. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. S5A-C) (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p17-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-use-db-dir-if-needed-fig-s5e-timing-1min">P17.  Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC(use –db_dir if needed) (Fig. S5E) (TIMING 1min)</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_wk.html">Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_em.html">Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_co_assem.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_subsample.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_assem_free.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">dbCAN-new-README</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/run_from_raw_reads_pr.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="run-from-raw-reads-priest-2023-automated-cazyme-and-glycan-substrate-annotation-in-microbiomes-a-step-by-step-protocol">
<h1>Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol<a class="headerlink" href="#run-from-raw-reads-priest-2023-automated-cazyme-and-glycan-substrate-annotation-in-microbiomes-a-step-by-step-protocol" title="Permalink to this heading"></a></h1>
<section id="example-3-priest2023-dataset-priest-2023">
<h2>Example 3: Priest2023 Dataset <a class="reference external" href="https://www.nature.com/articles/s43705-023-00324-7">priest_2023</a><a class="headerlink" href="#example-3-priest2023-dataset-priest-2023" title="Permalink to this heading"></a></h2>
<p>The Priest2023 dataset <a class="reference external" href="https://www.nature.com/articles/s43705-023-00324-7">priest_2023</a> was published in 2023 from a study of glycan utilization in the particulate and dissolved organic matter pools (POM and DOM) in the ocean waters of the Arctic during late summer1. This study applied PacBio HiFi long read sequencing for metagenomes (MGs) of 8 samples (4 sites of two depth) and Illumina short read sequencing for metatranscriptomes (MTs) of 4 samples (4 sites). It investigated CAZyme abundance and transcription profiles as well as inferred glycan degradation activities by comparing MGs and MTs from samples of different sites and depth. The sites are categorized into above-slope, above-shelf and open-ocean groups, and the water depth includes surface water (SRF) and bottom of the surface mixed layer (BML). CAZyme abundance and expression showed different profiles for the degradation of complex algal glycans such as sulfated fucan, laminarin and xylan in different site groups and water depths.</p>
<p>For this protocol, we will select two MG samples (HiFi DNA long reads, FRAM_WSC20_S25_SRF_MG and FRAM_WSC20_S25_BML_MG) and two MT samples (FRAM_WSC20_S25_SRF_MT and FRAM_WSC20_S25_BML_MT, paired-end 2x151bp reads) for CAZyme annotation. We will assemble HiFi long MG reads into contigs, which will be used to predict CAZymes and glycan substrates. We will map the MT reads to the contigs to calculate expression levels.
Procedure
———</p>
<section id="software-and-versions">
<h3>Software and versions<a class="headerlink" href="#software-and-versions" title="Permalink to this heading"></a></h3>
<ul class="simple">
<li><p><strong>Anaconda</strong> (<a class="reference external" href="https://www.anaconda.com">Anaconda</a>, version 23.7.3)</p></li>
<li><p><strong>Flye</strong> (<a class="reference external" href="https://github.com/mikolmogorov/Flye">Flye</a>, version 2.9.5)</p></li>
<li><p><strong>pyHMMER</strong> (<a class="reference external" href="https://pyhmmer.readthedocs.io/en/stable/">pyHMMER</a>, newest version)</p></li>
<li><p><strong>DIAMOND</strong> (<a class="reference external" href="https://github.com/bbuchfink/diamond">DIAMOND</a>, version 2.1.8)</p></li>
<li><p><strong>TrimGalore</strong> (<a class="reference external" href="https://github.com/FelixKrueger/TrimGalore">TrimGalore</a>, version 0.6.0)</p></li>
<li><p><strong>Pyrodigal</strong> (<a class="reference external" href="https://pyrodigal.readthedocs.io/en/stable/">Pyrodigal</a>, newest version)</p></li>
<li><p><strong>BWA</strong> (<a class="reference external" href="https://github.com/lh3/bwa">BWA</a>, version 0.7.17)</p></li>
<li><p><strong>Samtools</strong> (<a class="reference external" href="https://github.com/samtools/samtools">Samtools</a>, version 1.7)</p></li>
<li><p><strong>Kraken2</strong> (<a class="reference external" href="https://ccb.jhu.edu/software/kraken2/">Kraken2</a>, version 2.1.1)</p></li>
<li><p><strong>KrakenTools</strong> (<a class="reference external" href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file">KrakenTools</a>, version 1.2)</p></li>
<li><p><strong>run_dbcan</strong> (<a class="reference external" href="https://github.com/Xinpeng021001/run_dbCAN_new">run_dbcan</a>, version 5.0.3)</p></li>
<li><p><strong>MMseqs2</strong> (<a class="reference external" href="https://github.com/soedinglab/MMseqs2">MMseqs2</a>, release 15-6f452)</p></li>
<li><p><strong>Pysam</strong> (<a class="reference external" href="https://github.com/pysam-developers/pysam">Pysam</a>, version 0.22.1)</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>CAZyme_annotation_pr.yml
conda<span class="w"> </span>activate<span class="w"> </span>CAZyme_annotation_pr
</pre></div>
</div>
<p>To install the databases, execute the following commands:</p>
<section id="database-installation-command">
<h4>Database Installation Command<a class="headerlink" href="#database-installation-command" title="Permalink to this heading"></a></h4>
<p>prepare the conda environment (available at <a class="reference external" href="https://github.com/bcb-unl/run_dbcan_new/tree/master/envs">https://github.com/bcb-unl/run_dbcan_new/tree/master/envs</a>):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yml
conda<span class="w"> </span>activate<span class="w"> </span>run_dbcan_env
</pre></div>
</div>
<ol class="arabic simple">
<li><p>We provide command in the run_dbcan script:</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>database<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Users could also download all  database files from the dbCAN2 website (<a class="reference external" href="http://bcb.unl.edu/dbCAN2/download/Databases/">http://bcb.unl.edu/dbCAN2/download/Databases/</a>), and then put them into the db directory.</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/run_dbCAN_database_total/dbCAN_db.tar.gz<span class="w"> </span>-O<span class="w"> </span>db.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>db.tar.gz
rm<span class="w"> </span>db.tar.gz
</pre></div>
</div>
</section>
</section>
<section id="module-1-reads-processing-fig-3-to-obtain-contigs">
<h3>Module 1: Reads processing (Fig. 3) to obtain contigs<a class="headerlink" href="#module-1-reads-processing-fig-3-to-obtain-contigs" title="Permalink to this heading"></a></h3>
<section id="p1-contamination-check-timing-10min">
<h4>P1. Contamination Check (TIMING ~10min)<a class="headerlink" href="#p1-contamination-check-timing-10min" title="Permalink to this heading"></a></h4>
<p>Download the Priest2023 dataset:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/BML_MG.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/SRF_MG.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/BML_MT_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/BML_MT_2.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/SRF_MT_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/SRF_MT_2.fastq.gz
</pre></div>
</div>
<p>Use <cite>kraken2</cite> to check for contaminated reads:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>SRF_MT.kreport<span class="w"> </span>--output<span class="w"> </span>SRF_MT.kraken.output<span class="w"> </span>SRF_MT_1.fastq.gz<span class="w"> </span>SRF_MT_2.fastq.gz
kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>BML_MT.kreport<span class="w"> </span>--output<span class="w"> </span>BML_MT.kraken.output<span class="w"> </span>BML_MT_1.fastq.gz<span class="w"> </span>BML_MT_2.fastq.gz

kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>SRF_MG.kreport<span class="w"> </span>--output<span class="w"> </span>SRF_MG.kraken.output<span class="w"> </span>SRF_MG.fastq.gz
kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>BML_MG.kreport<span class="w"> </span>--output<span class="w"> </span>BML_MG.kraken.output<span class="w"> </span>BML_MG.fastq.gz
</pre></div>
</div>
<p>Kraken2 found much contamination (Box 1) from human in the Priest2023 data. Consequently, human reads need to be removed before assembly.</p>
<p>Reads can be aligned to the reference genomes of potential contamination source organisms to remove the aligned reads. The most common source in microbiome studies is from human.</p>
<section id="box-1-example-of-kraken2-output-files">
<h5>Box 1: Example of Kraken2 output files<a class="headerlink" href="#box-1-example-of-kraken2-output-files" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>The <cite>kreport</cite> files can be examined to identify potential contamination source organisms.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span>54M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:42<span class="w"> </span>BML_MG.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.2M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:42<span class="w"> </span>BML_MG.kreport
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">3</span>.4G<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">08</span>:01<span class="w"> </span>BML_MT.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>1023K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">08</span>:02<span class="w"> </span>BML_MT.kreport
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span>61M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:39<span class="w"> </span>SRF_MG.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.2M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:39<span class="w"> </span>SRF_MG.kreport
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">2</span>.6G<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:50<span class="w"> </span>SRF_MT.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.1M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">07</span>:51<span class="w"> </span>SRF_MT.kreport
</pre></div>
</div>
</div></blockquote>
</section>
</section>
<section id="p2-remove-contamination-reads-from-human-timing-40min">
<h4>P2. Remove contamination reads from human (TIMING ~40min)<a class="headerlink" href="#p2-remove-contamination-reads-from-human-timing-40min" title="Permalink to this heading"></a></h4>
<p>From the Kraken2 output files, we identified humans as the contamination source, we can use the following commands to remove the contamination reads by aligning reads to the human reference genome.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>hg38<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>hg38<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>wget<span class="w"> </span>https://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>mkdir<span class="w"> </span>contamination<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>contamination
minimap2<span class="w"> </span>-a<span class="w"> </span>-x<span class="w"> </span>map-hifi<span class="w"> </span>-MD<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>SRF_MG.hg38.sam<span class="w"> </span>../hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz<span class="w"> </span>../SRF_MG.fastq.gz
minimap2<span class="w"> </span>-a<span class="w"> </span>-x<span class="w"> </span>map-hifi<span class="w"> </span>-MD<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>BML_MG.hg38.sam<span class="w"> </span>../hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz<span class="w"> </span>../BML_MG.fastq.gz
samtools<span class="w"> </span>fastq<span class="w"> </span>-f<span class="w"> </span><span class="m">4</span><span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-0<span class="w"> </span>../SRF_MG.clean.fq.gz<span class="w"> </span>SRF_MG.hg38.sam
samtools<span class="w"> </span>fastq<span class="w"> </span>-f<span class="w"> </span><span class="m">4</span><span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-0<span class="w"> </span>../BML_MG.clean.fq.gz<span class="w"> </span>BML_MG.hg38.sam
bwa<span class="w"> </span>mem<span class="w"> </span>../hg38/hg38<span class="w"> </span>../SRF_MT_1.fastq.gz<span class="w"> </span>../SRF_MT_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>SRF_MT.hg38.sam
bwa<span class="w"> </span>mem<span class="w"> </span>../hg38/hg38<span class="w"> </span>../BML_MT_1.fastq.gz<span class="w"> </span>../BML_MT_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>BML_MT.hg38.sam
samtools<span class="w"> </span>fastq<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-1<span class="w"> </span>../SRF_MT_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>../SRF_MT_2.clean.fq.gz<span class="w"> </span>SRF_MT.hg38.sam
samtools<span class="w"> </span>fastq<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-1<span class="w"> </span>../BML_MT_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>../BML_MT_2.clean.fq.gz<span class="w"> </span>BML_MT.hg38.sam
<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>Or we could use KrakenTools to extract reads.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>SRF_MT.kraken.output<span class="w"> </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s1<span class="w"> </span>SRF_MT_1.fastq.gz<span class="w"> </span>-s2<span class="w"> </span>SRF_MT_2.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRF_MT_1.clean.fastq<span class="w"> </span>-o2<span class="w"> </span>SRF_MT_2.clean.fastq
gzip<span class="w"> </span>SRF_MT_1.clean.fastq
gzip<span class="w"> </span>SRF_MT_2.clean.fastq

extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>BML_MT.kraken.output<span class="w"> </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s1<span class="w"> </span>BML_MT_1.fastq.gz<span class="w"> </span>-s2<span class="w"> </span>BML_MT_2.fastq.gz<span class="w"> </span>-o<span class="w"> </span>BML_MT_1.clean.fastq<span class="w"> </span>-o2<span class="w"> </span>BML_MT_2.clean.fastq
gzip<span class="w"> </span>BML_MT_1.clean.fastq
gzip<span class="w"> </span>BML_MT_2.clean.fastq

extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>BML_MG.kraken.output<span class="w"> </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s<span class="w"> </span>BML_MG_.fastq.gz<span class="w"> </span>-o<span class="w"> </span>BML_MG_.clean.fastq
gzip<span class="w"> </span>BML_MG.clean.fastq

extract_kraken_reads.py<span class="w"> </span>-k<span class="w"> </span>SRF_MG.kraken.output<span class="w"> </span>-taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span>-exclude<span class="w"> </span>-s<span class="w"> </span>SRF_MG.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRF_MG.clean.fastq
gzip<span class="w"> </span>SRF_MG.clean.fastq
</pre></div>
</div>
</section>
<section id="p3-trim-adapter-and-low-quality-reads-timing-20min">
<h4>P3| Trim adapter and low-quality reads (TIMING ~20min)<a class="headerlink" href="#p3-trim-adapter-and-low-quality-reads-timing-20min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>trim_galore<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">8</span><span class="w"> </span>--paired<span class="w"> </span>BML_MT_1.clean.fastq.gz<span class="w"> </span>BML_MT_2.clean.fastq.gz
trim_galore<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">8</span><span class="w"> </span>--paired<span class="w"> </span>SRF_MT_1.clean.fastq.gz<span class="w"> </span>SRF_MT_2.clean.fastq.gz
</pre></div>
</div>
<p>The HiFi long reads do not need to be trimmed. Hence, this step only applies to MT illumina short read data. We specified –illumina to indicate that the reads were generated using the Illumina sequencing platform. Nonetheless, trim_galore possesses the ability to automatically detect the adapter, providing flexibility in adapter handling for users who may know the specific sequencing platform. Details of trimming are available in the trimming report file (Box 2).</p>
<section id="box-2-example-output-of-trim-galore">
<h5>Box 2: Example output of trim_galore<a class="headerlink" href="#box-2-example-output-of-trim-galore" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In addition to the trimmed read files, Trim_galore also generates a trimming report file. The trimming report contains details on read trimming, such as the number of trimmed reads.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.2K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">21</span>:56<span class="w"> </span>BML_MT_1.clean.fq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.3G<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">22</span>:05<span class="w"> </span>BML_MT_1.clean_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.7K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">22</span>:05<span class="w"> </span>BML_MT_2.clean.fq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.0G<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">22</span>:05<span class="w"> </span>BML_MT_2.clean_val_2.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.9K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:07<span class="w"> </span>SRF_MT_1.clean.fq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.7G<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:19<span class="w"> </span>SRF_MT_1.clean_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">5</span>.1K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:19<span class="w"> </span>SRF_MT_2.clean.fq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.3G<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:19<span class="w"> </span>SRF_MT_2.clean_val_2.fq.gz
</pre></div>
</div>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>During the trimming process, certain reads may be entirely removed due to low quality in its entirety. Using the <cite>–retain_unpaired</cite> parameter in trim_galore allows for the preservation of single-end reads. In this protocol, this option was not selected, so that both reads of a forward-revise pair were removed.</p>
</div>
</section>
</section>
<section id="p4-assemble-hifi-reads-into-metagenome-timing-4h20min">
<h4>P4. Assemble HiFi reads into metagenome (TIMING ~4h20min)<a class="headerlink" href="#p4-assemble-hifi-reads-into-metagenome-timing-4h20min" title="Permalink to this heading"></a></h4>
<p>Flye was used to assemble the HiFi long reads into contigs.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>flye<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--meta<span class="w"> </span>--pacbio-hifi<span class="w"> </span>BML_MG.clean.fq.gz<span class="w"> </span>--hifi-error<span class="w"> </span><span class="m">0</span>.01<span class="w"> </span>--keep-haplotypes<span class="w"> </span>--out-dir<span class="w"> </span>flye_BML_MG
flye<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--meta<span class="w"> </span>--pacbio-hifi<span class="w"> </span>SRF_MG.clean.fq.gz<span class="w"> </span>--hifi-error<span class="w"> </span><span class="m">0</span>.01<span class="w"> </span>--keep-haplotypes<span class="w"> </span>--out-dir<span class="w"> </span>flye_SRF_MG
</pre></div>
</div>
<p>Flye generates two folders <cite>flye_BML_MG</cite> and <cite>flye_SRF_MG</cite>. Each folder
contains 6 files and 5 sub-folders (Box 3), among them <cite>assembly.fasta</cite> is the final contig sequence file.
We set <cite>–hifi-error</cite> 0.01, a generally accepted error rate of HiFi sequencing.
Parameter <cite>–meta</cite> is set to assemble reads into metagenomes.</p>
<section id="box-3-example-output-of-flye">
<h5>Box 3: Example output of Flye<a class="headerlink" href="#box-3-example-output-of-flye" title="Permalink to this heading"></a></h5>
<blockquote>
<div><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">20</span>:15<span class="w"> </span><span class="m">00</span>-assembly
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">20</span>:43<span class="w"> </span><span class="m">10</span>-consensus
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">21</span>:14<span class="w"> </span><span class="m">20</span>-repeat
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">21</span>:16<span class="w"> </span><span class="m">30</span>-contigger
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span><span class="m">40</span>-polishing
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>314M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>assembly.fasta
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>311M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>assembly_graph.gfa
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">6</span>.6M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>assembly_graph.gv
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>867K<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>assembly_info.txt
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>61M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>flye.log
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span><span class="m">92</span><span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>params.json
</pre></div>
</div>
</div></blockquote>
</section>
</section>
<section id="p5-predict-genes-by-prodigal-14min">
<h4>P5. Predict genes by Prodigal (~14min)<a class="headerlink" href="#p5-predict-genes-by-prodigal-14min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pyrodigal<span class="w"> </span>-p<span class="w"> </span>meta<span class="w"> </span>-i<span class="w"> </span>flye_BML_MG/assembly.fasta<span class="w"> </span>-d<span class="w"> </span>BML_MG.cds<span class="w"> </span>-a<span class="w"> </span>BML_MG.faa<span class="w"> </span>-f<span class="w"> </span>gff<span class="w"> </span>-o<span class="w"> </span>BML_MG.gff<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
pyrodigal<span class="w"> </span>-p<span class="w"> </span>meta<span class="w"> </span>-i<span class="w"> </span>flye_SRF_MG/assembly.fasta<span class="w"> </span>-d<span class="w"> </span>SRF_MG.cds<span class="w"> </span>-a<span class="w"> </span>SRF_MG.faa<span class="w"> </span>-f<span class="w"> </span>gff<span class="w"> </span>-o<span class="w"> </span>SRF_MG.gff<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<section id="box-3-example-output-of-prokka">
<h5>Box 3: Example output of Prokka<a class="headerlink" href="#box-3-example-output-of-prokka" title="Permalink to this heading"></a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>105M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">23</span>:26<span class="w"> </span>BML_MG.faa
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>314M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">22</span>:06<span class="w"> </span>BML_MG.cds
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>467M<span class="w"> </span>Dec<span class="w"> </span><span class="m">27</span><span class="w"> </span><span class="m">23</span>:26<span class="w"> </span>BML_MG.gff
</pre></div>
</div>
</section>
</section>
</section>
<section id="module-1-run-dbcan-annotation-to-obtain-cazymes-cgcs-and-substrates">
<h3>Module 1: run_dbcan annotation to obtain CAZymes, CGCs, and substrates<a class="headerlink" href="#module-1-run-dbcan-annotation-to-obtain-cazymes-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<p>Hint: If you apply <code class="docutils literal notranslate"><span class="pre">prodigal</span></code> not <code class="docutils literal notranslate"><span class="pre">pyrodigal</span></code>, use this to fix gene ID in gff files by dbcan_utils (TIMING ~1min)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_utils<span class="w"> </span>gff_fix<span class="w"> </span>-i<span class="w"> </span>BML_MG.faa<span class="w"> </span>-g<span class="w"> </span>BML_MG.gff
dbcan_utils<span class="w"> </span>gff_fix<span class="w"> </span>-i<span class="w"> </span>SRF_MG.faa<span class="w"> </span>-g<span class="w"> </span>SRF_MG.gff
</pre></div>
</div>
<p>Users can skip P6 and P7, and directly run P8 (much slower though), if they want to predict not only CAZymes and CGCs, but also substrates.</p>
<section id="p6-cazyme-annotation-at-family-level-timing-10min">
<h4>P6. CAZyme annotation at family level (TIMING ~10min)<a class="headerlink" href="#p6-cazyme-annotation-at-family-level-timing-10min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>BML_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>BML_MG.CAZyme<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI
run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>SRF_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>SRF_MG.CAZyme<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w"> </span>--input_format<span class="w"> </span>NCBI
</pre></div>
</div>
<p>Two arguments are required for <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code>: the input sequence file (faa files) and the sequence type (protein).
By default, <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> will use three methods (<code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN</span> <span class="pre">HMMdb</span></code>, <code class="docutils literal notranslate"><span class="pre">DIAMOND</span></code> vs <code class="docutils literal notranslate"><span class="pre">CAZy</span></code>, <code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN-sub</span> <span class="pre">HMMdb</span></code>) for
CAZyme annotation.</p>
<p>The sequence type can be <cite>protein</cite>, <cite>prok</cite>, <cite>meta</cite>. If the input sequence file contains metagenomic contig sequences (<cite>fna</cite> file),
the sequence type has to be <cite>meta</cite>, and <cite>prodigal</cite> will be called to predict genes.</p>
</section>
<section id="p7-cgc-prediction-timing-15-min">
<h4>P7. CGC prediction (TIMING ~15 min)<a class="headerlink" href="#p7-cgc-prediction-timing-15-min" title="Permalink to this heading"></a></h4>
<p>The following commands will re-run run_dbcan to not only predict CAZymes but also CGCs with protein <cite>faa</cite> and gene location <cite>gff</cite> files.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>BML_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>BML_MG.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>BML_MG.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>SRF_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>SRF_MG.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>SRF_MG.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p><strong>Creating own gff file</strong>
If the users would like to create their own <code class="docutils literal notranslate"><span class="pre">gff</span></code> file (instead of using Prokka or Prodigal),
it is important to make sure the value of ID attribute in the <code class="docutils literal notranslate"><span class="pre">gff</span></code> file matches the protein ID in the protein <code class="docutils literal notranslate"><span class="pre">faa</span></code> file.</p>
<p><strong>[Troubleshooting]CGC not found</strong>
If no result is found in CGC output file, it is most likely because the sequence IDs in <code class="docutils literal notranslate"><span class="pre">gff</span></code> file and <code class="docutils literal notranslate"><span class="pre">faa</span></code> file do not match.
Another less likely reason is that the contigs are too short and fragmented and not suitable for CGC prediction.</p>
</div>
</section>
<section id="p8-substrate-prediction-for-cazymes-and-cgcs-timing-5h">
<h4>P8. Substrate prediction for CAZymes and CGCs (TIMING ~5h)<a class="headerlink" href="#p8-substrate-prediction-for-cazymes-and-cgcs-timing-5h" title="Permalink to this heading"></a></h4>
<p>The following commands will re-run run_dbcan to predict CAZymes, CGCs.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>BML_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>BML_MG.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>BML_MG.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>SRF_MG.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w">  </span>--output_dir<span class="w"> </span>SRF_MG.PUL<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--input_gff<span class="w"> </span>SRF_MG.gff<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<section id="box-6-example-output-folder-content-of-run-dbcan-substrate-prediction">
<h5>Box 6: Example output folder content of run_dbcan substrate prediction<a class="headerlink" href="#box-6-example-output-folder-content-of-run-dbcan-substrate-prediction" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In the output directory (<a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/BML_MG.dbCAN/">https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/BML_MG.dbCAN/</a>), a total of 17 files and 1 folder are generated:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">9</span>.6M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>PUL_blast.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.8M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>CGC.faa
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span>26M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>cgc.gff
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>450K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>cgc.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>212K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>cgc_standard.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>1005K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>cgc_standard.out.json
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>406K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:11<span class="w"> </span>dbcan-sub.hmm.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>325K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:11<span class="w"> </span>diamond.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>332K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:11<span class="w"> </span>dtemp.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>220K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:11<span class="w"> </span>hmmer.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>240K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>overview.txt
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.7M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:17<span class="w"> </span>stp.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span>17K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:18<span class="w"> </span>substrate.out
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">   </span>12K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:19<span class="w"> </span>synteny.pdf
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>293K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:13<span class="w"> </span>tf-1.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>222K<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:15<span class="w"> </span>tf-2.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">1</span>.7M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">10</span>:17<span class="w"> </span>tp.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>105M<span class="w"> </span>Dec<span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">05</span>:57<span class="w"> </span>uniInput
</pre></div>
</div>
</div></blockquote>
<p>Descriptions of Output Files:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">PUL_blast.out</span></code>: BLAST results between CGCs and PULs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC.faa</span></code>: CGC Fasta sequences.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cgc.gff</span></code>: reformatted from the user input gff file by marking CAZymes, TFs, TCs, and STPs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cgc.out</span></code>: raw output of CGC predictions.</p></li>
</ul>
</div></blockquote>
<p>Each entry in cgc.out includes:</p>
<ol class="arabic simple">
<li><p>CGC_id: CGC1</p></li>
<li><p>type: CAZyme</p></li>
<li><p>contig_id: contig_10157</p></li>
<li><p>gene_id: BML_MG_01992</p></li>
<li><p>start: 33003</p></li>
<li><p>end: 36077</p></li>
<li><p>strand: +</p></li>
<li><p>annotation: GH2</p></li>
</ol>
<p>Explanation: the gene BML_MG_01992 encodes a GH2 CAZyme in the CGC1 of the contig contig_10157. CGC1 also has other genes, which are provided in other rows. BML_MG_01992 is on the positive strand of contig_10157 from 33003 to 36077. The type can be one of the four signature gene types (CAZymes, TCs, TFs, STPs) or the null type (not annotated as one of the four signature genes).</p>
<p><cite>cgc_standard.out.json</cite>: JSON format of cgc_standard.out.
<cite>dbcan-sub.hmm.out</cite>: HMMER search result against dbCAN-sub HMMdb, including a column with CAZyme substrates extracted from fam-substrate-mapping-08012023.tsv.
<cite>diamond.out</cite>: DIAMOND search result against the CAZy annotated protein sequences (CAZyDB.07262023.fa).
<cite>dtemp.out</cite>: temporary file.
<cite>hmmer.out</cite>: HMMER search result against dbCAN HMMdb.
<cite>overview.txt</cite>: summary of CAZyme annotation from three methods in TSV format. An example row has the following columns:
1.      Gene_ID: BML_MG_01761
2.      EC#: 2.4.99.-:5
3.      dbCAN: GT112(19-370)
4.      dbCAN_sub: GT112_e0
5.      DIAMOND: GT112
6.      #ofTools: 3
Explanation: the protein BML_MG_01761 is annotated by 3 tools to be a CAZyme: (1) GT112 (CAZy defined family GT112) by HMMER vs dbCAN HMMdb with a domain range from aa position 19 to 370, (2) GT112_e0 (eCAMI defined subfamily e0; e indicates it is from eCAMI not CAZy) by HMMER vs dbCAN-sub HMMdb (derived from eCAMI subfamilies), and (3) GT112 by DIAMOND vs CAZy annotated protein sequences. The second column 2.4.99.-:5 is extracted from eCAMI, meaning that the eCAMI subfamily GT112_e0 contains 5 member proteins which have an EC 2.4.99.- according to CAZy. In most cases, the 3 tools will have the same CAZyme family assignment. When they give different assignment. We recommend a preference order: dbCAN &gt; eCAMI/dbCAN-sub &gt; DIAMOND. See our dbCAN2 paper2, dbCAN3 paper3, and eCAMI4 for more details.
Note: If users invoked the –use_signalP parameter when running run_dbcan, there will be an additional column called signal in the overview.txt.
stp.out: HMMER search result against the MiST5 compiled signal transduction protein HMMs from Pfam.
tf-1.out: HMMER search result against the DBD6 compiled transcription factor HMMs from Pfam 7.
tf-2.out: HMMER search result against the DBD compiled transcription factor HMMs from Superfamily 8.
tp.out: DIAMOND search result against the TCDB 9 annotated protein sequences.
substrate.out: summary of substrate prediction results for CGCs in TSV format from two approaches3 (dbCAN-PUL blast search and dbCAN-sub majority voting). An example row has the following columns:
1.      CGC_ID: contig_10778|CGC2
2.      Best hit PUL_ID in dbCAN-PUL: PUL0400
3.      Substrate of the hit PUL: alginate
4.      Sum of bitscores for homologous gene pairs between CGC and PUL: 851.0
5.      Types of homologous gene pairs: CAZyme-CAZyme;CAZyme-CAZyme;CAZyme-CAZyme;CAZyme-CAZyme
6.      Substrate predicted by majority voting of CAZymes in CGC: alginate
7.      Voting score: 2.0
Explanation: The CGC2 of contig_10778 has its best hit PUL0400 (from PUL_blast.out) with alginate as substrate (from dbCAN-PUL_12-12-2023.xlsx). Four signature genes are matched between contig_10778|CGC2 and PUL0400 (from PUL_blast.out): all the four are CAZymes. The sum of blast bitscores of the 4 homologous pairs (CAZyme-CAZyme;CAZyme-CAZyme;CAZyme-CAZyme;CAZyme-CAZyme) is 851.0. Hence, the substrate of contig_10778|CGC2 is predicted to be alginate according to dbCAN-PUL blast search. The last two columns are based on the dbCAN-sub result (dbcan-sub.hmm.out), according to which two CAZymes in contig_10778|CGC2 are predicted to have alginate substrate. The voting score is thus 2.0, so that according to the majority voting rule, contig_10778|CGC2 is predicted to have an alginate substrate.
Note: for many CGCs, only one of the two approaches produces substrate prediction. In some cases, the two approaches produce different substrate assignments. We recommend a preference order: dbCAN-PUL blast search &gt; dbCAN-sub majority voting. See our dbCAN3 paper3 for more details.
synteny.pdf: a folder with syntenic block alignment plots between all CGCs and PULs.
uniInput: renamed Fasta file from input protein sequence file.</p>
</section>
</section>
</section>
<section id="module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">
<h3>Module 3. Read mapping (Fig. 3) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates<a class="headerlink" href="#module-3-read-mapping-fig-3-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<section id="p9-read-mapping-to-all-contigs-of-each-sample-timing-10-min">
<h4>P9. Read mapping to all contigs of each sample (TIMING ~10 min)<a class="headerlink" href="#p9-read-mapping-to-all-contigs-of-each-sample-timing-10-min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>samfiles
bwa<span class="w"> </span>index<span class="w"> </span>flye_BML_MG/assembly.fasta
bwa<span class="w"> </span>index<span class="w"> </span>flye_SRF_MG/assembly.fasta

bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/HCZGU.sam<span class="w"> </span>flye_BML_MG/assembly.fasta<span class="w"> </span>HCZGU_1_val_1.fq.gz<span class="w"> </span>HCZGU_2_val_2.fq.gz
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/HTOOS.sam<span class="w"> </span>flye_SRF_MG/assembly.fasta<span class="w"> </span>HTOOS_1_val_1.fq.gz<span class="w"> </span>HTOOS_2_val_2.fq.gz

bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/BML_MT.sam<span class="w"> </span>flye_BML_MG/assembly.fasta<span class="w"> </span>BML_MT_1.clean_val_1.fq.gz<span class="w"> </span>BML_MT_2.clean_val_2.fq.gz
bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/SRF_MT.sam<span class="w"> </span>flye_SRF_MG/assembly.fasta<span class="w"> </span>SRF_MT_1.clean_val_1.fq.gz<span class="w"> </span>SRF_MT_2.clean_val_2.fq.gz
</pre></div>
</div>
</section>
<section id="p10-hifi-mg-long-read-mapping-to-all-contigs-of-each-sample-timing-20min">
<h4>P10. HiFi MG long read mapping to all contigs of each sample (TIMING ~20min)<a class="headerlink" href="#p10-hifi-mg-long-read-mapping-to-all-contigs-of-each-sample-timing-20min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>minimap2<span class="w"> </span>-a<span class="w"> </span>-x<span class="w"> </span>map-hifi<span class="w"> </span>--MD<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/BML_MG.sam<span class="w"> </span>flye_BML_MG/assembly.fasta<span class="w"> </span>BML_MG.clean.fq.gz

minimap2<span class="w"> </span>-a<span class="w"> </span>-x<span class="w"> </span>map-hifi<span class="w"> </span>--MD<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/SRF_MG.sam<span class="w"> </span>flye_SRF_MG/assembly.fasta<span class="w"> </span>SRF_MG.clean.fq.gz
</pre></div>
</div>
<p>The –MD parameter is required in this command, because step P12 will access the MD tag in SAM files generated in this step to calculate the coverage.</p>
<p>P11. Convert SAM to BAM and sort (TIMING ~5min)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>samfiles

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>SRF_MG.bam<span class="w"> </span>SRF_MG.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>BML_MG.bam<span class="w"> </span>BML_MG.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>BML_MT.bam<span class="w"> </span>BML_MT.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>SRF_MT.bam<span class="w"> </span>SRF_MT.sam

samtools<span class="w"> </span>index<span class="w"> </span>BML_MT.bam

samtools<span class="w"> </span>index<span class="w"> </span>SRF_MT.bam

samtools<span class="w"> </span>index<span class="w"> </span>BML_MG.bam

samtools<span class="w"> </span>index<span class="w"> </span>SRF_MG.bam

rm<span class="w"> </span>-rf<span class="w"> </span>*sam

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
</section>
<section id="p12-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utilsbedtools-timing-6min12min">
<h4>P12. Read count calculation for all proteins of each sample using dbcan_utilsBedtools (TIMING ~6min12min)<a class="headerlink" href="#p12-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utilsbedtools-timing-6min12min" title="Permalink to this heading"></a></h4>
<p>Calculate HiFi MG read abundance for proteins in the two samples:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>SRF_MG_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>SRF_MG_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../SRF_MG.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/SRF_MG.bam<span class="w"> </span>-o<span class="w"> </span>SRF_MG.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>–-identity<span class="w"> </span><span class="m">0</span>.98<span class="w"> </span>--hifi

<span class="nb">cd</span><span class="w"> </span>..



mkdir<span class="w"> </span>BML_MG_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>BML_MG_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../BML_MG.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/BML_MG.bam<span class="w"> </span>-o<span class="w"> </span>BML_MG.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>–-identity<span class="w"> </span><span class="m">0</span>.98<span class="w"> </span>--hifi

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>Calculate Illumina MT read abundance for proteins in the two samples:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>BML_MT_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>BML_MT_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../BML_MG.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/BML_MT.bam<span class="w"> </span>-o<span class="w"> </span>BML_MT.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>–-identity<span class="w"> </span><span class="m">0</span>.98

<span class="nb">cd</span><span class="w"> </span>..



mkdir<span class="w"> </span>SRF_MT_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>SRF_MT_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../SRF_MG.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/SRF_MT.bam<span class="w"> </span>-o<span class="w"> </span>SRF_MT.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>–-identity<span class="w"> </span><span class="m">0</span>.98

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>Read counts are saved in depth.txt files of each sample.</p>
</section>
<section id="p13-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min">
<h4>P13. Read count calculation for a given region of contigs using Samtools (TIMING ~1min)<a class="headerlink" href="#p13-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>SRF_MG_abund

samtools<span class="w"> </span>depth<span class="w"> </span>-r<span class="w"> </span>contig_12725:10309-25115<span class="w"> </span>../samfiles/SRF_MT.bam<span class="w"> </span>&gt;<span class="w"> </span>SRF_MT.cgc.depth.txt

<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>SRF_MT_abund

samtools<span class="w"> </span>depth<span class="w"> </span>-r<span class="w"> </span>contig_12725:10309-25115<span class="w"> </span>../samfiles/SRF_MG.bam<span class="w"> </span>&gt;<span class="w"> </span>SRF_MG.cgc.depth.txt

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>The parameter <cite>-r contig_12725:10309-25115contig_6487:2406-6727`</cite> specifies a region in a contig. For any CGC, its positional range can be found in the file cgc_standard output produced by run_dbcan. The depth.txt files contain the raw read counts for the specified region.</p>
</section>
<section id="p14-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">
<h4>P14. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)<a class="headerlink" href="#p14-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min" title="Permalink to this heading"></a></h4>
<p>As the Priest2023 dataset has both MG and MT data, for each sample we will have two abundance folders, e.g., BML_MG_abund, BML_MT_abund for BML sample, and SRF_MG_abund, SRF_MT_abund for SRF sample.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>BML_MT_abund

dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM


<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>SRF_MT_abund

dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MT.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM


<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>BML_MG_abund

dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>BML_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../BML_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM


<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>SRF_MG_abund

dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>SRF_MG.depth.txt<span class="w"> </span>-i<span class="w"> </span>../SRF_MG.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>We developed a set of Python scripts as <code class="docutils literal notranslate"><span class="pre">dbcan_utils</span></code> (included in the <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> package) to take the raw read counts for all genes as input and output the normalized abundances (refer to Box 7) of CAZyme families, subfamilies, CGCs, and substrates (see Fig. 4). The parameter <code class="docutils literal notranslate"><span class="pre">-a</span> <span class="pre">TPM</span></code> can also be set to two other metrics: RPM, or RPKM61.</p>
<ul class="simple">
<li><p><strong>RPKM</strong> is calculated as the number of mapped reads to a gene G divided by [(total number of mapped reads to all genes / 10^6) x (gene G length / 1000)].</p></li>
<li><p><strong>RPM</strong> is the number of mapped reads to a gene G divided by (total number of mapped reads to all genes / 10^6).</p></li>
<li><p><strong>TPM</strong> is calculated as [number of mapped reads to a gene G / (gene G length / 1000)] divided by the sum of [number of mapped reads to each gene / (the gene length / 1000)].</p></li>
</ul>
<section id="box-7-example-output-of-dbcan-utils">
<h5>Box 7. Example output of dbcan_utils<a class="headerlink" href="#box-7-example-output-of-dbcan-utils" title="Permalink to this heading"></a></h5>
<p>As an example, <cite>SRF_MG_abund  &lt;https://bcb.unl.edu/dbCAN_tutorial/dataset3-Priest2023/SRF_MG_abund/&gt;_</cite> folder has 7 TSV files:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>122K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>CGC_abund.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">835</span><span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>CGC_substrate_majority_voting.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.5K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>CGC_substrate_PUL_homology.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.6K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>EC_abund.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.3K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>fam_abund.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>18K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>fam_substrate_abund.out

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">6</span>.9M<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">04</span>:07<span class="w"> </span>SRF_MG.depth.txt

-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>19K<span class="w"> </span>Jun<span class="w"> </span><span class="m">19</span><span class="w"> </span><span class="m">07</span>:30<span class="w"> </span>subfam_abund.out
</pre></div>
</div>
<p>Explanation of columns in these TSV files is as follows:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">fam_abund.out</span></code>: CAZy family (from HMMER vs dbCAN HMMdb), sum of TPM, number of CAZymes in the family.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">subfam_abund.out</span></code>: eCAMI subfamily (from HMMER vs dbCAN-sub HMMdb), sum of TPM, number of CAZymes in the subfamily.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">EC_abund.out</span></code>: EC number (extracted from dbCAN-sub subfamily), sum of TPM, number of CAZymes with the EC.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fam_substrate_abund.out</span></code>: Substrate (from HMMER vs dbCAN-sub HMMdb), sum of TPM (all CAZymes in this substrate group), GeneID (all CAZyme IDs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_abund.out</span></code>: CGC_ID (e.g., k141_338400|CGC1), mean of TPM (all genes in the CGC), Seq_IDs (IDs of all genes in the CGC), TPM (of all genes in the CGC), Families (CAZyme family or other signature gene type of all genes in the CGC).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_PUL_homology.out</span></code>: Substrate (from dbCAN-PUL blast search), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-PUL blast search), TPM (of CGCs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_majority_voting.out</span></code>: Substrate (from dbCAN-sub majority voting), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-sub majority voting), TPM (of CGCs in this substrate group).</p></li>
</ul>
</div></blockquote>
</section>
</section>
</section>
<section id="module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">
<h3>Module 4: dbcan_plot for data visualization (Fig. 3) of abundances of CAZymes, CGCs, and substrates (TIMING variable)<a class="headerlink" href="#module-4-dbcan-plot-for-data-visualization-fig-3-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable" title="Permalink to this heading"></a></h3>
<section id="p15-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min">
<h4>P15. Heatmap for CAZyme substrate abundance across samples (TIMING 1min)<a class="headerlink" href="#p15-heatmap-for-cazyme-substrate-abundance-across-samples-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>heatmap_plot<span class="w"> </span>--show_abund<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>--samples<span class="w"> </span>BML_MG,BML_MT,SRF_MG,SRF_MT<span class="w"> </span>-i<span class="w"> </span>BML_MG_abund/fam_substrate_abund.out,BML_MT_abund/fam_substrate_abund.out,SRF_MG_abund/fam_substrate_abund.out,SRF_MT_abund/fam_substrate_abund.out
</pre></div>
</div>
</section>
<section id="p16-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s5a-c-timing-1min">
<h4>P16. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. S5A-C) (TIMING 1min)<a class="headerlink" href="#p16-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-s5a-c-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>BML_MG,BML_MT,SRF_MG,SRF_MT<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>BML_MG_abund/fam_abund.out,BML_MT_abund/fam_abund.out,SRF_MG_abund/fam_abund.out,SRF_MT_abund/fam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>fam.pdf

dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>BML_MG,BML_MT,SRF_MG,SRF_MT<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>BML_MG_abund/subfam_abund.out,BML_MT_abund/subfam_abund.out,SRF_MG_abund/subfam_abund.out,SRF_MT_abund/subfam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>subfam.pdf

dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>BML_MG,BML_MT,SRF_MG,SRF_MT<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>BML_MG_abund/EC_abund.out,BML_MT_abund/EC_abund.out,SRF_MG_abund/EC_abund.out,SRF_MT_abund/EC_abund.out<span class="w"> </span>--pdf<span class="w"> </span>ec.pdf
</pre></div>
</div>
</section>
<section id="p17-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-use-db-dir-if-needed-fig-s5e-timing-1min">
<h4>P17.  Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC(use –db_dir if needed) (Fig. S5E) (TIMING 1min)<a class="headerlink" href="#p17-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-use-db-dir-if-needed-fig-s5e-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_coverage_plot<span class="w"> </span>-i<span class="w"> </span>SRF_MG.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;contig_12725|CGC1&#39;</span><span class="w"> </span>--readscount<span class="w"> </span>SRF_MG_abund/SRF_MG.cgc.depth.txt

dbcan_plot<span class="w"> </span>CGC_synteny_coverage_plot<span class="w"> </span>-i<span class="w"> </span>SRF_MG.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;contig_12725|CGC1&#39;</span><span class="w"> </span>--readscount<span class="w"> </span>SRF_MT_abund/SRF_MT.cgc.depth.txt
</pre></div>
</div>
<p>If users only want to plot the CGC structure:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_plot<span class="w"> </span>-i<span class="w"> </span>SRF_MG.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;contig_12725|CGC1&#39;</span>
</pre></div>
</div>
<p>If users only want to plot the CGC structure plus the read mapping coverage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_coverage_plot<span class="w"> </span>-i<span class="w"> </span>SRF_MG.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;contig_12725|CGC1&#39;</span><span class="w"> </span>--readscount<span class="w"> </span>SRF_MG_abund/SRF_MG.cgc.depth.txt
</pre></div>
</div>
<p>If users only want to plot the synteny between the CGC and PUL:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_plot<span class="w"> </span>-i<span class="w"> </span>SRF_MG.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;contig_12725|CGC1&#39;</span>
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="run_from_raw_reads_am.html" class="btn btn-neutral float-left" title="Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="run_from_raw_reads_wk.html" class="btn btn-neutral float-right" title="Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Xinpeng Zhang; Haidong Yi; Jinfang Zheng.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>