

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free &mdash; dbCAN-new-README 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=87f11076"></script>
      <script src="_static/doctools.js?v=888ff710"></script>
      <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample" href="run_from_raw_reads_sp_subsample.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            dbCAN-new-README
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="prepare_the_database.html">Database Installation Command</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_fasta_data_processing.html">Process the input fasta file</a></li>
<li class="toctree-l1"><a class="reference internal" href="CAZyme_annotation.html">CAZyme Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_information_generation.html">Prepare the CGC annotation information</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_annotation.html">Run with CGCFinder</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_CGC_substrate.html">Substrate prediction with CGCs</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_substrate_plots.html">CGC substrate visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads.html">Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_am.html">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_pr.html">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_wk.html">Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_em.html">Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_co_assem.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_subsample.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#overview">Overview</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#example-carter-2023-dataset-carter-2023">Example : Carter 2023 Dataset <code class="docutils literal notranslate"><span class="pre">Carter</span> <span class="pre">2023</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#s1-download-carter2023-raw-reads-10min">S1. Download Carter2023 raw reads (~10min)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#s2-install-anaconda-3min">S2. Install Anaconda (~3min)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#s4-configure-databases-required-by-run-dbcan-2h">S4. Configure databases required by run_dbcan (~2h)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#database-installation-command">Database Installation Command</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#procedure">Procedure</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#module-1-reads-processing-fig-2-to-obtain-contigs">Module 1: Reads processing (Fig. 2) to obtain contigs</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p1-contamination-check">P1. Contamination Check</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p2-trim-adapter-and-low-quality-reads-timing-20min">P2. Trim adapter and low-quality reads (TIMING ~20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p3-covert-the-fastq-file-to-fasta-9min">P3. Covert the fastq file to fasta (~9min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-2-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">Module 2. Read mapping (Fig. 2) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p4-map-reads-to-proteins-in-cazydb-by-using-diamond-blastx-5h10min">P4. Map reads to proteins in CAZyDB by using DIAMOND blastx (~5h10min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p5-dbcan-asmfree-to-calculate-the-abundance-of-cazyme-families-subfamily-ec-and-substrate-11min">P5. dbcan_asmfree to calculate the abundance of CAZyme families, subfamily, EC and substrate (~11min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">Module 4: dbcan_plot for data visualization (Fig. 2) of abundances of CAZymes, CGCs, and substrates (TIMING variable)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p6-barplot-for-cazyme-family-subfamily-and-ec-abundance-across-samples-timing-1min">P6. Barplot for CAZyme family, subfamily, and EC abundance across samples (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p7-heatmap-for-cazyme-substrate-abundance-across-samples-fig-s2d-timing-1min">P7. Heatmap for CAZyme substrate abundance across samples (Fig. S2D) (TIMING 1min)</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">dbCAN-new-README</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/run_from_raw_reads_sp_assem_free.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="run-from-raw-reads-cater-2023-supplementary-protocol-for-assembly-free">
<h1>Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free<a class="headerlink" href="#run-from-raw-reads-cater-2023-supplementary-protocol-for-assembly-free" title="Permalink to this heading"></a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this heading"></a></h2>
<section id="overview">
<h3>Overview<a class="headerlink" href="#overview" title="Permalink to this heading"></a></h3>
<p>There are three sections in this document:</p>
<p>Procedure of <cite>assembly-free</cite></p>
</section>
</section>
<section id="example-carter-2023-dataset-carter-2023">
<h2>Example : Carter 2023 Dataset <code class="docutils literal notranslate"><span class="pre">Carter</span> <span class="pre">2023</span></code><a class="headerlink" href="#example-carter-2023-dataset-carter-2023" title="Permalink to this heading"></a></h2>
<section id="s1-download-carter2023-raw-reads-10min">
<h3>S1. Download Carter2023 raw reads (~10min)<a class="headerlink" href="#s1-download-carter2023-raw-reads-10min" title="Permalink to this heading"></a></h3>
<p>To download the required raw reads, use the following wget commands:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Dry2014_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Dry2014_2.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_2.fastq.gz
</pre></div>
</div>
<p>These raw data were originally downloaded from
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=ERR7745896">https://www.ncbi.nlm.nih.gov/sra/?term=ERR7745896</a>
and <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=ERR7738162">https://www.ncbi.nlm.nih.gov/sra/?term=ERR7738162</a>
and renamed to indicate their collected seasons.</p>
</section>
<section id="s2-install-anaconda-3min">
<h3>S2. Install Anaconda (~3min)<a class="headerlink" href="#s2-install-anaconda-3min" title="Permalink to this heading"></a></h3>
<p>Download and install the latest version of Anaconda for Linux from
<a class="reference external" href="https://www.anaconda.com/download#downloads">https://www.anaconda.com/download#downloads</a>. Once Anaconda is
successfully installed, proceed to create a dedicated conda environment
named <code class="docutils literal notranslate"><span class="pre">CAZyme_annotation</span></code> and activate it.
Subsequently, all the required tools can be seamlessly installed within
this environment (yml file is provided on github).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>CAZyme_annotation_main.yml
conda<span class="w"> </span>activate<span class="w"> </span>CAZyme_annotation_main
</pre></div>
</div>
</section>
<section id="s4-configure-databases-required-by-run-dbcan-2h">
<h3>S4. Configure databases required by run_dbcan (~2h)<a class="headerlink" href="#s4-configure-databases-required-by-run-dbcan-2h" title="Permalink to this heading"></a></h3>
<p>To install the databases, execute the following commands:</p>
<section id="database-installation-command">
<h4>Database Installation Command<a class="headerlink" href="#database-installation-command" title="Permalink to this heading"></a></h4>
<p>prepare the conda environment (available at <a class="reference external" href="https://github.com/bcb-unl/run_dbcan_new/tree/master/envs">https://github.com/bcb-unl/run_dbcan_new/tree/master/envs</a>):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yml
conda<span class="w"> </span>activate<span class="w"> </span>run_dbcan_env
</pre></div>
</div>
<ol class="arabic simple">
<li><p>We provide command in the run_dbcan script:</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>database<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Users could also download all  database files from the dbCAN2 website (<a class="reference external" href="http://bcb.unl.edu/dbCAN2/download/Databases/">http://bcb.unl.edu/dbCAN2/download/Databases/</a>), and then put them into the db directory.</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/dbCAN_db.tar.gz<span class="w"> </span>-O<span class="w"> </span>db.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>db.tar.gz
rm<span class="w"> </span>db.tar.gz
</pre></div>
</div>
<p>Download database required by Kraken2 (very slow; can be skipped
if users do not intend to run Kraken2):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2-build<span class="w"> </span>--standard<span class="w"> </span>--db<span class="w"> </span>K2
</pre></div>
</div>
<p>or could download database directly from the <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">https://benlangmead.github.io/aws-indexes/k2</a>,
and extract the tarball to the db folder, here we use standard database as an example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://genome-idx.s3.amazonaws.com/kraken/k2_standard_20240605.tar.gz
mkdir<span class="w"> </span>K2
tar<span class="w"> </span>-xvf<span class="w"> </span>k2_standard_20240605.tar.gz<span class="w"> </span>-C<span class="w"> </span>K2
</pre></div>
</div>
<p><strong>CRITICAL STEP</strong></p>
<blockquote>
<div><p>The downloaded files must be all in the right location (the db folder).</p>
<p>The CAZy.dmnd file is needed for DIAMOND search.</p>
<p>The dbCAN.hmm and dbCAN_sub.hmm files are for HMMER search.</p>
<p>The tcdb.dmnd, TF.HMM, and STP.hmm files are for CGC prediction.</p>
<p>The PUL.faa file consists of protein sequences from experimentally
validated PULs for BLAST search to predict substrates for CGCs.</p>
<p>The dbCAN-PUL_12-12-2023.txt and dbCAN-PUL_12-12-2023.xlsx files contain
PUL-substrate mapping curated from literature.</p>
<p>Lastly, the
fam-substrate-mapping.tsv file is the family-EC-substrate
mapping table for the prediction of CAZyme substrates.</p>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Users should use a clean version of Anaconda. If the above steps failed, we suggest users reinstall their Anaconda.
The Anaconda installation and configuration step may experience
prolonged time while resolving environment dependencies.
Users should be patient during this process. Alternatively,
users may consider “mamba”, another Python package manager
that offers similar functionality to Anaconda. Information and
access to mamba software can be found at
<a class="reference external" href="https://github.com/mamba-org/mamba">https://github.com/mamba-org/mamba</a>.</p>
</div>
</section>
</section>
</section>
<section id="procedure">
<h2>Procedure<a class="headerlink" href="#procedure" title="Permalink to this heading"></a></h2>
<section id="module-1-reads-processing-fig-2-to-obtain-contigs">
<h3>Module 1: Reads processing (Fig. 2) to obtain contigs<a class="headerlink" href="#module-1-reads-processing-fig-2-to-obtain-contigs" title="Permalink to this heading"></a></h3>
<section id="p1-contamination-check">
<h4>P1. Contamination Check<a class="headerlink" href="#p1-contamination-check" title="Permalink to this heading"></a></h4>
<p>Use <cite>kraken2</cite> to check for contaminated reads:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>Wet2014.kreport<span class="w"> </span>--output<span class="w"> </span>Wet2014.kraken.output<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz
kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>Dry2014.kreport<span class="w"> </span>--output<span class="w"> </span>Dry2014.kraken.output<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz
</pre></div>
</div>
<p>Kraken2 found very little contamination in the Carter2023 data. Consequently, there was no need for the contamination removal step.
For Kraken2 visualization, users can use pavian (<a class="reference external" href="https://github.com/fbreitwieser/pavian?tab=readme-ov-file">https://github.com/fbreitwieser/pavian?tab=readme-ov-file</a>).</p>
<p>If contamination is identified, users can align the reads to the reference genomes of potential contamination source organisms to remove
the aligned reads (Box 1). The most common source in human microbiome studies is from human hosts.</p>
<section id="box-1-removing-contamination-reads-from-humans">
<h5>Box 1: Removing Contamination Reads from Humans<a class="headerlink" href="#box-1-removing-contamination-reads-from-humans" title="Permalink to this heading"></a></h5>
<p>Kraken2 will produce the following output files.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.0G<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">10</span>:24<span class="w"> </span>Dry2014.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">1</span>.2M<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">10</span>:25<span class="w"> </span>Dry2014.kreport
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">5</span>.1G<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">09</span>:47<span class="w"> </span>Wet2014.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">1</span>.1M<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">09</span>:48<span class="w"> </span>Wet2014.kreport
</pre></div>
</div>
<p>Suppose from these files, we have identified humans as the contamination source, we can use the following commands to remove the contamination reads by aligning reads to the human reference genome.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>index<span class="w"> </span>-p<span class="w"> </span>hg38<span class="w"> </span>Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Wet2014.hg38.sam
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Dry2014.hg38.sam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>Wet2014.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>Wet2014.hg38.unmap.bam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>Dry2014.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>Dry2014.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>Wet2014_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>Wet2014_2.clean.fq.gz<span class="w"> </span>Wet2014.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>Dry2014_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>Dry2014_2.clean.fq.gz<span class="w"> </span>Dry2014.hg38.unmap.bam
</pre></div>
</div>
<p>KrakenTools could also extract host reads quickly and easied which is recommended. We use tax 2759 (plant) as an example.
Please read KrakenTools README for more information (<a class="reference external" href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file">https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file</a>).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>extract_kraken_reads.py<span class="w"> </span><span class="se">\</span>
-k<span class="w"> </span>Dry2014.kraken.output<span class="w"> </span><span class="se">\</span>
-s1<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>-s2<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span><span class="se">\</span>
--fastq-output<span class="w"> </span>--exclude<span class="w"> </span><span class="se">\</span>
--taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span><span class="se">\</span>
-o<span class="w"> </span>Dry2014_1.clean.fq.gz<span class="w"> </span>-o2<span class="w"> </span>Dry2014_2.clean.fq.gz
</pre></div>
</div>
</section>
</section>
<section id="p2-trim-adapter-and-low-quality-reads-timing-20min">
<h4>P2. Trim adapter and low-quality reads (TIMING ~20min)<a class="headerlink" href="#p2-trim-adapter-and-low-quality-reads-timing-20min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>trim_galore<span class="w"> </span>--paired<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
trim_galore<span class="w"> </span>--paired<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<p>We specified <cite>–illumina</cite> to indicate that the reads were generated using the Illumina sequencing platform.
Nonetheless, trim_galore can automatically detect adapters, providing flexibility for users who may know the specific sequencing platform.
Details of trimming are available in the trimming report file (Box 2).</p>
<section id="box-2-example-output-of-trim-galore">
<h5>Box 2: Example output of <cite>trim_galore</cite><a class="headerlink" href="#box-2-example-output-of-trim-galore" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In addition to the trimmed read files, <cite>Trim_galore`</cite> also generates a trimming report file.
The trimming report contains details on read trimming, such as the number of trimmed reads.</p>
</div></blockquote>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.2K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:48<span class="w"> </span>Dry2014_1.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.0G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_1_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.4K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_2.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.4G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_2_val_2.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.4K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:30<span class="w"> </span>Wet2014_1.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.4G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_1_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.6K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_2.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.7G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_2_val_2.fq.gz
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>During the trimming process, certain reads may be entirely removed due to low quality in its entirety.
Using the <code class="docutils literal notranslate"><span class="pre">--retain_unpaired</span></code> parameter in <code class="docutils literal notranslate"><span class="pre">trim_galore</span></code> allows for the preservation of single-end reads.
In this protocol, this option was not selected, so that both reads of a forward-revise pair were removed.</p>
</div>
</section>
</section>
<section id="p3-covert-the-fastq-file-to-fasta-9min">
<h4>P3. Covert the fastq file to fasta (~9min)<a class="headerlink" href="#p3-covert-the-fastq-file-to-fasta-9min" title="Permalink to this heading"></a></h4>
<p>Need to install <cite>seqtk`</cite> first</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>seqtk<span class="w"> </span>seq<span class="w"> </span>-a<span class="w"> </span>Dry2014_1_val_1.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>Dry2014_1.fna

seqtk<span class="w"> </span>seq<span class="w"> </span>-a<span class="w"> </span>Dry2014_2_val_2.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>Dry2014_2.fna

seqtk<span class="w"> </span>seq<span class="w"> </span>-a<span class="w"> </span>Wet2014_1_val_1.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>Wet2014_1.fna

seqtk<span class="w"> </span>seq<span class="w"> </span>-a<span class="w"> </span>Wet2014_2_val_2.fq.gz<span class="w"> </span>&gt;<span class="w"> </span>Wet2014_2.fna
</pre></div>
</div>
</section>
</section>
<section id="module-2-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">
<h3>Module 2. Read mapping (Fig. 2) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates<a class="headerlink" href="#module-2-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<section id="p4-map-reads-to-proteins-in-cazydb-by-using-diamond-blastx-5h10min">
<h4>P4. Map reads to proteins in CAZyDB by using DIAMOND blastx (~5h10min)<a class="headerlink" href="#p4-map-reads-to-proteins-in-cazydb-by-using-diamond-blastx-5h10min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>ln<span class="w"> </span>-s<span class="w"> </span>db/CAZy.dmnd<span class="w"> </span>.

diamond<span class="w"> </span>blastx<span class="w"> </span>--db<span class="w"> </span>CAZy<span class="w"> </span>--out<span class="w"> </span>Dry2014_1.blastx<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>qseqid<span class="w"> </span>sseqid<span class="w"> </span>pident<span class="w"> </span>length<span class="w"> </span>mismatch<span class="w"> </span>gapopen<span class="w"> </span>qstart<span class="w"> </span>qend<span class="w"> </span>sstart<span class="w"> </span>send<span class="w"> </span>evalue<span class="w"> </span>bitscore<span class="w"> </span>qcovhsp<span class="w"> </span>qlen<span class="w"> </span>slen<span class="w"> </span>--id<span class="w"> </span><span class="m">80</span><span class="w"> </span>--query-cover<span class="w"> </span><span class="m">90</span><span class="w"> </span>--query<span class="w"> </span>Dry2014_1.fna<span class="w"> </span>--max-target-seqs<span class="w"> </span><span class="m">1</span><span class="w"> </span>--quiet

diamond<span class="w"> </span>blastx<span class="w"> </span>--db<span class="w"> </span>CAZy<span class="w"> </span>--out<span class="w"> </span>Dry2014_2.blastx<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>qseqid<span class="w"> </span>sseqid<span class="w"> </span>pident<span class="w"> </span>length<span class="w"> </span>mismatch<span class="w"> </span>gapopen<span class="w"> </span>qstart<span class="w"> </span>qend<span class="w"> </span>sstart<span class="w"> </span>send<span class="w"> </span>evalue<span class="w"> </span>bitscore<span class="w"> </span>qcovhsp<span class="w"> </span>qlen<span class="w"> </span>slen<span class="w"> </span>--id<span class="w"> </span><span class="m">80</span><span class="w"> </span>--query-cover<span class="w"> </span><span class="m">90</span><span class="w"> </span>--query<span class="w"> </span>Dry2014_2.fna<span class="w"> </span>--max-target-seqs<span class="w"> </span><span class="m">1</span><span class="w"> </span>--quiet

diamond<span class="w"> </span>blastx<span class="w"> </span>--db<span class="w"> </span>CAZy<span class="w"> </span>--out<span class="w"> </span>Wet2014_1.blastx<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>qseqid<span class="w"> </span>sseqid<span class="w"> </span>pident<span class="w"> </span>length<span class="w"> </span>mismatch<span class="w"> </span>gapopen<span class="w"> </span>qstart<span class="w"> </span>qend<span class="w"> </span>sstart<span class="w"> </span>send<span class="w"> </span>evalue<span class="w"> </span>bitscore<span class="w"> </span>qcovhsp<span class="w"> </span>qlen<span class="w"> </span>slen<span class="w"> </span>--id<span class="w"> </span><span class="m">80</span><span class="w"> </span>--query-cover<span class="w"> </span><span class="m">90</span><span class="w"> </span>--query<span class="w"> </span>Wet2014_1.fna<span class="w"> </span>--max-target-seqs<span class="w"> </span><span class="m">1</span><span class="w"> </span>--quiet

diamond<span class="w"> </span>blastx<span class="w"> </span>--db<span class="w"> </span>CAZy<span class="w"> </span>--out<span class="w"> </span>Wet2014_2.blastx<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>qseqid<span class="w"> </span>sseqid<span class="w"> </span>pident<span class="w"> </span>length<span class="w"> </span>mismatch<span class="w"> </span>gapopen<span class="w"> </span>qstart<span class="w"> </span>qend<span class="w"> </span>sstart<span class="w"> </span>send<span class="w"> </span>evalue<span class="w"> </span>bitscore<span class="w"> </span>qcovhsp<span class="w"> </span>qlen<span class="w"> </span>slen<span class="w"> </span>--id<span class="w"> </span><span class="m">80</span><span class="w"> </span>--query-cover<span class="w"> </span><span class="m">90</span><span class="w"> </span>--query<span class="w"> </span>Wet2014_2.fna<span class="w"> </span>--max-target-seqs<span class="w"> </span><span class="m">1</span><span class="w"> </span>--quiet
</pre></div>
</div>
<p>The parameters <cite>–query-cover 90`</cite> and <cite>–id 80`</cite> are required to align the reads to CAZy database. The thresholds of these parameters will significantly impact the results. The <cite>–max-target-seqs 1`</cite> parameter specify to only keep the best CAZy hit, which means each read will be mapped to only one CAZyme protein in the CAZyDB.</p>
</section>
<section id="p5-dbcan-asmfree-to-calculate-the-abundance-of-cazyme-families-subfamily-ec-and-substrate-11min">
<h4>P5. dbcan_asmfree to calculate the abundance of CAZyme families, subfamily, EC and substrate (~11min)<a class="headerlink" href="#p5-dbcan-asmfree-to-calculate-the-abundance-of-cazyme-families-subfamily-ec-and-substrate-11min" title="Permalink to this heading"></a></h4>
<p>We developed a Python script as dbcan_asmfree (included in the run_dbcan package) to take the DIAMOND blastx results from pair-end reads and the raw reads as inputs. With the diamond_fam_abund method, it will output the CAZyme family abundance, with the diamond_subfam_abund method, it will output the eCAMI subfamily abundance, with the diamond_EC_abund method, it will output the EC abundance and with the diamond_substrate_abund method, it will output the substrate abundance.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_asmfree<span class="w"> </span>diamond_fam_abund<span class="w"> </span>-paf1<span class="w"> </span>Dry2014_1.blastx<span class="w"> </span>-paf2<span class="w"> </span>Dry2014_2.blastx<span class="w"> </span>--raw_reads<span class="w"> </span>Dry2014_1_val_1.fq.gz<span class="w">  </span>-n<span class="w"> </span>FPKM<span class="w"> </span>-o<span class="w"> </span>Dry2014_fam_abund

dbcan_asmfree<span class="w"> </span>diamond_fam_abund<span class="w"> </span>-paf1<span class="w"> </span>Wet2014_1.blastx<span class="w"> </span>-paf2<span class="w"> </span>Wet2014_2.blastx<span class="w"> </span>--raw_reads<span class="w"> </span>Wet2014_1_val_1.fq.gz<span class="w"> </span>-n<span class="w"> </span>FPKM<span class="w"> </span>-o<span class="w"> </span>Wet2014_fam_abund



dbcan_asmfree<span class="w"> </span>diamond_subfam_abund<span class="w"> </span>-paf1<span class="w"> </span>Dry2014_1.blastx<span class="w"> </span>-paf2<span class="w"> </span>Dry2014_2.blastx<span class="w"> </span>--raw_reads<span class="w"> </span>Dry2014_1_val_1.fq.gz<span class="w"> </span>-o<span class="w"> </span>Dry2014_subfam_abund<span class="w"> </span>-n<span class="w"> </span>FPKM

dbcan_asmfree<span class="w"> </span>diamond_subfam_abund<span class="w"> </span>-paf1<span class="w"> </span>Wet2014_1.blastx<span class="w"> </span>-paf2<span class="w"> </span>Wet2014_2.blastx<span class="w"> </span>--raw_reads<span class="w"> </span>Wet2014_1_val_1.fq.gz<span class="w"> </span>-o<span class="w"> </span>Wet2014_subfam_abund<span class="w"> </span>-n<span class="w"> </span>FPKM



dbcan_asmfree<span class="w"> </span>diamond_EC_abund<span class="w"> </span>-i<span class="w"> </span>Dry2014_subfam_abund<span class="w"> </span>-o<span class="w"> </span>Dry2014_EC_abund

dbcan_asmfree<span class="w"> </span>diamond_EC_abund<span class="w"> </span>-i<span class="w"> </span>Wet2014_subfam_abund<span class="w"> </span>-o<span class="w"> </span>Wet2014_EC_abund



dbcan_asmfree<span class="w"> </span>diamond_substrate_abund<span class="w"> </span>-i<span class="w"> </span>Dry2014_subfam_abund<span class="w"> </span>-o<span class="w"> </span>Dry2014_substrate_abund

dbcan_asmfree<span class="w"> </span>diamond_substrate_abund<span class="w"> </span>-i<span class="w"> </span>Wet2014_subfam_abund<span class="w"> </span>-o<span class="w"> </span>Wet2014_substrate_abund
</pre></div>
</div>
<p>The substrate prediction in the assemble-free route is based on the fam-substrate-mapping-08012023.tsv mapping table. Specifically, from the blastx result files, reads are first grouped according to their best CAZyDB proteins. Then, reads are further grouped according to the CAZyme families and eCAMI subfamilies where the best CAZyDB proteins belong to. Lastly, the fam-substrate-mapping-08012023.tsv mapping table is used to map CAZyme families and ECs (from eCAMI subfamilies) to substrates.
Therefore, the substrate abundance is calculated in a similar way as implemented in dbCAN-sub.</p>
</section>
</section>
<section id="module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">
<h3>Module 4: dbcan_plot for data visualization (Fig. 2) of abundances of CAZymes, CGCs, and substrates (TIMING variable)<a class="headerlink" href="#module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable" title="Permalink to this heading"></a></h3>
<section id="p6-barplot-for-cazyme-family-subfamily-and-ec-abundance-across-samples-timing-1min">
<h4>P6. Barplot for CAZyme family, subfamily, and EC abundance across samples (TIMING 1min)<a class="headerlink" href="#p6-barplot-for-cazyme-family-subfamily-and-ec-abundance-across-samples-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_fam_abund,Dry2014_fam_abund<span class="w"> </span>--pdf<span class="w"> </span>fam.pdfDry2014.blastx.CAZy.FPKM.tsv,Wet2014.blastx.CAZy.FPKM.tsv

dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_subfam_abund,Dry2014_subfam_abund<span class="w"> </span>--pdf<span class="w"> </span>subfam.pdf

dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_fam_abund,Dry2014_fam_abund<span class="w"> </span>--pdf<span class="w"> </span>ec.pdf
</pre></div>
</div>
<p>Here we plot the top 20 substrates in the two samples. The input files are the two CAZyme substrate abundance files calculated based on
dbCAN-sub result. The default heatmap is ranked by substrate abundances. To rank the heatmap according to abundance profile using
the clustermap function of the seaborn package (<a class="reference external" href="https://github.com/mwaskom/seaborn">https://github.com/mwaskom/seaborn</a>), users can invoke the <code class="docutils literal notranslate"><span class="pre">--cluster_map</span></code> parameter.</p>
</section>
<section id="p7-heatmap-for-cazyme-substrate-abundance-across-samples-fig-s2d-timing-1min">
<h4>P7. Heatmap for CAZyme substrate abundance across samples (Fig. S2D) (TIMING 1min)<a class="headerlink" href="#p7-heatmap-for-cazyme-substrate-abundance-across-samples-fig-s2d-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>heatmap_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--show_abund<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w">  </span>Wet2014_substrate_abund,Dry2014_substrate_abund
</pre></div>
</div>
</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="run_from_raw_reads_sp_subsample.html" class="btn btn-neutral float-left" title="Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Xinpeng Zhang; Haidong Yi; Jinfang Zheng.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>