

<!DOCTYPE html>
<html class="writer-html5" lang="en">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly &mdash; dbCAN-new-README 0.1.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=87f11076"></script>
      <script src="_static/doctools.js?v=888ff710"></script>
      <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample" href="run_from_raw_reads_sp_subsample.html" />
    <link rel="prev" title="Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" href="run_from_raw_reads_em.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            dbCAN-new-README
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="prepare_the_database.html">Database Installation Command</a></li>
<li class="toctree-l1"><a class="reference internal" href="quick_start.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_fasta_data_processing.html">Process the input fasta file</a></li>
<li class="toctree-l1"><a class="reference internal" href="CAZyme_annotation.html">CAZyme Annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_information_generation.html">Prepare the CGC annotation information</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_annotation.html">Run with CGCFinder</a></li>
<li class="toctree-l1"><a class="reference internal" href="predict_CGC_substrate.html">Substrate prediction with CGCs</a></li>
<li class="toctree-l1"><a class="reference internal" href="CGC_substrate_plots.html">CGC substrate visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads.html">Run from Raw Reads(Cater 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_am.html">Run from Raw Reads(Amelia 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_pr.html">Run from Raw Reads(Priest 2023): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_wk.html">Run from Raw Reads(Wastyk 2021): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_em.html">Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#overview">Overview</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#example-carter-2023-dataset-carter-2023">Example : Carter 2023 Dataset <code class="docutils literal notranslate"><span class="pre">Carter</span> <span class="pre">2023</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#s1-download-carter2023-raw-reads-10min">S1. Download Carter2023 raw reads (~10min)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#s2-install-anaconda-3min">S2. Install Anaconda (~3min)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#s4-configure-databases-required-by-run-dbcan-2h">S4. Configure databases required by run_dbcan (~2h)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#database-installation-command">Database Installation Command</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#procedure">Procedure</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#module-1-reads-processing-fig-2-to-obtain-contigs">Module 1: Reads processing (Fig. 2) to obtain contigs</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p1-contamination-check">P1. Contamination Check</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p2-trim-adapter-and-low-quality-reads-timing-20min">P2. Trim adapter and low-quality reads (TIMING ~20min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p3-assemble-reads-into-contigs">P3. Assemble reads into contigs</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p4-predict-genes-by-prodigal-timing-21min">P4. Predict genes by <cite>Prodigal</cite> (TIMING ~21min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-2-run-dbcan-annotation-fig-2-to-obtain-cazymes-cgcs-and-substrates">Module 2. run_dbcan annotation (Fig. 2) to obtain CAZymes, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min">P5. CAZyme annotation at the CAZyme family level (TIMING ~10min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p6-cgc-prediction-timing-15-min">P6. CGC prediction (TIMING ~15 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h">P7. Substrate prediction for CAZymes and CGCs (TIMING ~5h)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-3-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">Module 3. Read mapping (Fig. 2) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p8-read-mapping-to-all-cds-of-each-sample-timing-20-min">P8. Read mapping to all CDS of each sample (TIMING ~20 min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p9-sort-sam-files-by-coordinates-timing-4min">P9. Sort SAM files by coordinates (TIMING ~4min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-12min">P10. Read count calculation for all proteins of each sample using dbcan_utils  (TIMING ~12min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-2min">P11. Read count calculation for a given region of contigs using Samtools (TIMING ~2min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">P12. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">Module 4: dbcan_plot for data visualization (Fig. 2) of abundances of CAZymes, CGCs, and substrates (TIMING variable)</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#p13-heatmap-for-cazyme-substrate-abundance-across-samples-fig-6a-timing-1min">P13. Heatmap for CAZyme substrate abundance across samples (Fig. 6A) (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-6b-c-timing-1min">P14. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. 6B,C) (TIMING 1min)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-6d-timing-1min">P15. Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC (Fig. 6D) (TIMING 1min)</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#troubleshooting">Troubleshooting</a></li>
<li class="toctree-l2"><a class="reference internal" href="#procedure-timing">Procedure Timing</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_subsample.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample</a></li>
<li class="toctree-l1"><a class="reference internal" href="run_from_raw_reads_sp_assem_free.html">Run from Raw Reads(Cater 2023): Supplementary Protocol for assembly-free</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">dbCAN-new-README</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/run_from_raw_reads_sp_co_assem.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="run-from-raw-reads-cater-2023-supplementary-protocol-for-co-assembly">
<h1>Run from Raw Reads(Cater 2023): Supplementary Protocol for co-assembly<a class="headerlink" href="#run-from-raw-reads-cater-2023-supplementary-protocol-for-co-assembly" title="Permalink to this heading"></a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this heading"></a></h2>
<section id="overview">
<h3>Overview<a class="headerlink" href="#overview" title="Permalink to this heading"></a></h3>
<p>There are three sections in this document:</p>
<p>Procedure of <cite>co-assembly</cite></p>
</section>
</section>
<section id="example-carter-2023-dataset-carter-2023">
<h2>Example : Carter 2023 Dataset <code class="docutils literal notranslate"><span class="pre">Carter</span> <span class="pre">2023</span></code><a class="headerlink" href="#example-carter-2023-dataset-carter-2023" title="Permalink to this heading"></a></h2>
<section id="s1-download-carter2023-raw-reads-10min">
<h3>S1. Download Carter2023 raw reads (~10min)<a class="headerlink" href="#s1-download-carter2023-raw-reads-10min" title="Permalink to this heading"></a></h3>
<p>To download the required raw reads, use the following wget commands:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Dry2014_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Dry2014_2.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_1.fastq.gz
wget<span class="w"> </span>https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_2.fastq.gz
</pre></div>
</div>
<p>These raw data were originally downloaded from
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=ERR7745896">https://www.ncbi.nlm.nih.gov/sra/?term=ERR7745896</a>
and <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/?term=ERR7738162">https://www.ncbi.nlm.nih.gov/sra/?term=ERR7738162</a>
and renamed to indicate their collected seasons.</p>
</section>
<section id="s2-install-anaconda-3min">
<h3>S2. Install Anaconda (~3min)<a class="headerlink" href="#s2-install-anaconda-3min" title="Permalink to this heading"></a></h3>
<p>Download and install the latest version of Anaconda for Linux from
<a class="reference external" href="https://www.anaconda.com/download#downloads">https://www.anaconda.com/download#downloads</a>. Once Anaconda is
successfully installed, proceed to create a dedicated conda environment
named <code class="docutils literal notranslate"><span class="pre">CAZyme_annotation</span></code> and activate it.
Subsequently, all the required tools can be seamlessly installed within
this environment (yml file is provided on github).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>CAZyme_annotation_main.yml
conda<span class="w"> </span>activate<span class="w"> </span>CAZyme_annotation_main
</pre></div>
</div>
</section>
<section id="s4-configure-databases-required-by-run-dbcan-2h">
<h3>S4. Configure databases required by run_dbcan (~2h)<a class="headerlink" href="#s4-configure-databases-required-by-run-dbcan-2h" title="Permalink to this heading"></a></h3>
<p>To install the databases, execute the following commands:</p>
<section id="database-installation-command">
<h4>Database Installation Command<a class="headerlink" href="#database-installation-command" title="Permalink to this heading"></a></h4>
<p>prepare the conda environment:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yml<span class="w"> </span><span class="o">(</span>available<span class="w"> </span>at<span class="w"> </span>https://github.com/Xinpeng021001/run_dbCAN_new/blob/master/environment.yml<span class="o">)</span>
conda<span class="w"> </span>activate<span class="w"> </span>dbCAN-test-env
pip<span class="w"> </span>install<span class="w"> </span>run-dbcan-new
</pre></div>
</div>
<ol class="arabic simple">
<li><p>We provide command in the run_dbcan script:</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>database<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Users could also download all  database files from the dbCAN2 website (<a class="reference external" href="http://bcb.unl.edu/dbCAN2/download/Databases/">http://bcb.unl.edu/dbCAN2/download/Databases/</a>), and then put them into the db directory.</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>-q<span class="w"> </span>https://bcb.unl.edu/dbCAN2/download/test/dbCAN_db.tar.gz<span class="w"> </span>-O<span class="w"> </span>db.tar.gz
tar<span class="w"> </span>-zxvf<span class="w"> </span>db.tar.gz
rm<span class="w"> </span>db.tar.gz
</pre></div>
</div>
<p>Download database required by Kraken2 (very slow; can be skipped
if users do not intend to run Kraken2):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2-build<span class="w"> </span>--standard<span class="w"> </span>--db<span class="w"> </span>K2
</pre></div>
</div>
<p>or could download database directly from the <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">https://benlangmead.github.io/aws-indexes/k2</a>,
and extract the tarball to the db folder, here we use standard database as an example:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://genome-idx.s3.amazonaws.com/kraken/k2_standard_20240605.tar.gz
mkdir<span class="w"> </span>K2
tar<span class="w"> </span>-xvf<span class="w"> </span>k2_standard_20240605.tar.gz<span class="w"> </span>-C<span class="w"> </span>K2
</pre></div>
</div>
<p><strong>CRITICAL STEP</strong></p>
<blockquote>
<div><p>The downloaded files must be all in the right location (the db folder).</p>
<p>The CAZy.dmnd file is needed for DIAMOND search.</p>
<p>The dbCAN.hmm and dbCAN_sub.hmm files are for HMMER search.</p>
<p>The tcdb.dmnd, TF.HMM, and STP.hmm files are for CGC prediction.</p>
<p>The PUL.faa file consists of protein sequences from experimentally
validated PULs for BLAST search to predict substrates for CGCs.</p>
<p>The dbCAN-PUL_12-12-2023.txt and dbCAN-PUL_12-12-2023.xlsx files contain
PUL-substrate mapping curated from literature.</p>
<p>Lastly, the
fam-substrate-mapping.tsv file is the family-EC-substrate
mapping table for the prediction of CAZyme substrates.</p>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Users should use a clean version of Anaconda. If the above steps failed, we suggest users reinstall their Anaconda.
The Anaconda installation and configuration step may experience
prolonged time while resolving environment dependencies.
Users should be patient during this process. Alternatively,
users may consider “mamba”, another Python package manager
that offers similar functionality to Anaconda. Information and
access to mamba software can be found at
<a class="reference external" href="https://github.com/mamba-org/mamba">https://github.com/mamba-org/mamba</a>.</p>
</div>
</section>
</section>
</section>
<section id="procedure">
<h2>Procedure<a class="headerlink" href="#procedure" title="Permalink to this heading"></a></h2>
<section id="module-1-reads-processing-fig-2-to-obtain-contigs">
<h3>Module 1: Reads processing (Fig. 2) to obtain contigs<a class="headerlink" href="#module-1-reads-processing-fig-2-to-obtain-contigs" title="Permalink to this heading"></a></h3>
<section id="p1-contamination-check">
<h4>P1. Contamination Check<a class="headerlink" href="#p1-contamination-check" title="Permalink to this heading"></a></h4>
<p>Use <cite>kraken2</cite> to check for contaminated reads:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>Wet2014.kreport<span class="w"> </span>--output<span class="w"> </span>Wet2014.kraken.output<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz
kraken2<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>--quick<span class="w"> </span>--paired<span class="w"> </span>--db<span class="w"> </span>K2<span class="w"> </span>--report<span class="w"> </span>Dry2014.kreport<span class="w"> </span>--output<span class="w"> </span>Dry2014.kraken.output<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz
</pre></div>
</div>
<p>Kraken2 found very little contamination in the Carter2023 data. Consequently, there was no need for the contamination removal step.
For Kraken2 visualization, users can use pavian (<a class="reference external" href="https://github.com/fbreitwieser/pavian?tab=readme-ov-file">https://github.com/fbreitwieser/pavian?tab=readme-ov-file</a>).</p>
<p>If contamination is identified, users can align the reads to the reference genomes of potential contamination source organisms to remove
the aligned reads (Box 1). The most common source in human microbiome studies is from human hosts.</p>
<section id="box-1-removing-contamination-reads-from-humans">
<h5>Box 1: Removing Contamination Reads from Humans<a class="headerlink" href="#box-1-removing-contamination-reads-from-humans" title="Permalink to this heading"></a></h5>
<p>Kraken2 will produce the following output files.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.0G<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">10</span>:24<span class="w"> </span>Dry2014.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">1</span>.2M<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">10</span>:25<span class="w"> </span>Dry2014.kreport
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">5</span>.1G<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">09</span>:47<span class="w"> </span>Wet2014.kraken.output
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">1</span>.1M<span class="w"> </span>Dec<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">09</span>:48<span class="w"> </span>Wet2014.kreport
</pre></div>
</div>
<p>Suppose from these files, we have identified humans as the contamination source, we can use the following commands to remove the contamination reads by aligning reads to the human reference genome.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>https://ftp.ensembl.org/pub/release-110/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>index<span class="w"> </span>-p<span class="w"> </span>hg38<span class="w"> </span>Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Wet2014.hg38.sam
bwa<span class="w"> </span>mem<span class="w"> </span>hg38<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Dry2014.hg38.sam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>Wet2014.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>Wet2014.hg38.unmap.bam
samtools<span class="w"> </span>view<span class="w"> </span>-f<span class="w"> </span><span class="m">12</span><span class="w"> </span>Dry2014.hg38.sam<span class="w"> </span>&gt;<span class="w"> </span>Dry2014.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>Wet2014_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>Wet2014_2.clean.fq.gz<span class="w"> </span>Wet2014.hg38.unmap.bam
samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>Dry2014_1.clean.fq.gz<span class="w"> </span>-2<span class="w"> </span>Dry2014_2.clean.fq.gz<span class="w"> </span>Dry2014.hg38.unmap.bam
</pre></div>
</div>
<p>KrakenTools could also extract host reads quickly and easied which is recommended. We use tax 2759 (plant) as an example.
Please read KrakenTools README for more information (<a class="reference external" href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file">https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file</a>).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>extract_kraken_reads.py<span class="w"> </span><span class="se">\</span>
-k<span class="w"> </span>Dry2014.kraken.output<span class="w"> </span><span class="se">\</span>
-s1<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>-s2<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span><span class="se">\</span>
--fastq-output<span class="w"> </span>--exclude<span class="w"> </span><span class="se">\</span>
--taxid<span class="w"> </span><span class="m">9606</span><span class="w"> </span><span class="se">\</span>
-o<span class="w"> </span>Dry2014_1.clean.fq.gz<span class="w"> </span>-o2<span class="w"> </span>Dry2014_2.clean.fq.gz
</pre></div>
</div>
</section>
</section>
<section id="p2-trim-adapter-and-low-quality-reads-timing-20min">
<h4>P2. Trim adapter and low-quality reads (TIMING ~20min)<a class="headerlink" href="#p2-trim-adapter-and-low-quality-reads-timing-20min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>trim_galore<span class="w"> </span>--paired<span class="w"> </span>Wet2014_1.fastq.gz<span class="w"> </span>Wet2014_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
trim_galore<span class="w"> </span>--paired<span class="w"> </span>Dry2014_1.fastq.gz<span class="w"> </span>Dry2014_2.fastq.gz<span class="w"> </span>--illumina<span class="w"> </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<p>We specified <cite>–illumina</cite> to indicate that the reads were generated using the Illumina sequencing platform.
Nonetheless, trim_galore can automatically detect adapters, providing flexibility for users who may know the specific sequencing platform.
Details of trimming are available in the trimming report file (Box 2).</p>
<section id="box-2-example-output-of-trim-galore">
<h5>Box 2: Example output of <cite>trim_galore</cite><a class="headerlink" href="#box-2-example-output-of-trim-galore" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In addition to the trimmed read files, <cite>Trim_galore`</cite> also generates a trimming report file.
The trimming report contains details on read trimming, such as the number of trimmed reads.</p>
</div></blockquote>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.2K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:48<span class="w"> </span>Dry2014_1.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.0G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_1_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.4K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_2.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.4G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:55<span class="w"> </span>Dry2014_2_val_2.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.4K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:30<span class="w"> </span>Wet2014_1.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.4G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_1_val_1.fq.gz
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.6K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_2.fastq.gz_trimming_report.txt
-rw-rw-r--<span class="w"> </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">3</span>.7G<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">01</span>:46<span class="w"> </span>Wet2014_2_val_2.fq.gz
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>During the trimming process, certain reads may be entirely removed due to low quality in its entirety.
Using the <code class="docutils literal notranslate"><span class="pre">--retain_unpaired</span></code> parameter in <code class="docutils literal notranslate"><span class="pre">trim_galore</span></code> allows for the preservation of single-end reads.
In this protocol, this option was not selected, so that both reads of a forward-revise pair were removed.</p>
</div>
</section>
</section>
<section id="p3-assemble-reads-into-contigs">
<h4>P3. Assemble reads into contigs<a class="headerlink" href="#p3-assemble-reads-into-contigs" title="Permalink to this heading"></a></h4>
<p>Use Megahit for assembling reads into contigs:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>zcat<span class="w"> </span>Dry2014_1_val_1.fq.gz<span class="w"> </span>Wet2014_1_val_1.fq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>gzip<span class="w"> </span>&gt;<span class="w"> </span>co2014_1.fq.gz

zcat<span class="w"> </span>Dry2014_2_val_1.fq.gz<span class="w"> </span>Wet2014_2_val_1.fq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>gzip<span class="w"> </span>&gt;<span class="w"> </span>co2014_2.fq.gz

megahit<span class="w"> </span>--min-contig-len<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-1<span class="w"> </span>co2014_1.fq.gz<span class="w"> </span>-2<span class="w"> </span>co2014_2.fq.gz<span class="w"> </span>-m<span class="w"> </span><span class="m">0</span>.5<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>megahit_Hadza_INDIV_157<span class="w"> </span>--out-prefix<span class="w"> </span>Hadza_INDIV_157
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">MEGAHIT</span></code> generates two output folders. Each contains five files and one sub-folder (Box 3).
<code class="docutils literal notranslate"><span class="pre">Wet2014.contigs.fa</span></code> is the final contig sequence file. We set <cite>–min-contig-len 1000</cite>,
a common practice to retain all contigs longer than 1,000 base pairs.</p>
<section id="box-3-example-output-of-megahit">
<h5>Box 3: Example output of <cite>MEGAHIT</cite><a class="headerlink" href="#box-3-example-output-of-megahit" title="Permalink to this heading"></a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span><span class="m">262</span><span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">04</span>:19<span class="w"> </span>checkpoints.txt
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">    </span><span class="m">0</span><span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">04</span>:19<span class="w"> </span><span class="k">done</span>
drwxrwxr-x<span class="w">  </span><span class="m">2</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.0K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">04</span>:19<span class="w"> </span>intermediate_contigs
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">1</span>.1K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">02</span>:22<span class="w"> </span>options.json
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>258M<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">04</span>:19<span class="w"> </span>Wet2014.contigs.fa
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>208K<span class="w"> </span>Dec<span class="w"> </span><span class="m">13</span><span class="w"> </span><span class="m">04</span>:19<span class="w"> </span>Wet2014.log
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>A common practice in metagenomics after assembly is to further bin contigs into metagenome-assembled genomes (MAGs).
However, in this protocol, we chose not to generate MAGs because not all contigs can be binned into MAGs, and those un-binned
contigs can also encode CAZymes.</p>
</div>
</section>
</section>
<section id="p4-predict-genes-by-prodigal-timing-21min">
<h4>P4. Predict genes by <cite>Prodigal</cite> (TIMING ~21min)<a class="headerlink" href="#p4-predict-genes-by-prodigal-timing-21min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pyrodigal<span class="w"> </span>-p<span class="w"> </span>meta<span class="w"> </span>-i<span class="w"> </span>megahit_Hadza_INDIV_157/Hadza_INDIV_157.contigs.fa<span class="w"> </span>-d<span class="w"> </span>Hadza_INDIV_157.cds<span class="w"> </span>-a<span class="w"> </span>Hadza_INDIV_157.faa<span class="w"> </span>-f<span class="w"> </span>gff<span class="w"> </span>-o<span class="w"> </span>Hadza_INDIV_157.gff<span class="w">  </span>-j<span class="w"> </span><span class="m">36</span>
</pre></div>
</div>
<p>The output files comprise of both protein and CDS sequences in Fasta format (e.g., Dry2014.faa and Wet2014.cds in Box 4).
The parameter -o and -f gff indicates the general feature format of gene annotation (e.g., Dry2014.gff).</p>
<section id="box-4-example-output-of-prodigal">
<h5>Box 4: Example output of <cite>Prodigal</cite><a class="headerlink" href="#box-4-example-output-of-prodigal" title="Permalink to this heading"></a></h5>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>215M<span class="w"> </span>May<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="m">10</span>:17<span class="w"> </span>Hadza_INDIV.faa
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>710M<span class="w"> </span>May<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="m">10</span>:17<span class="w"> </span>Hadza_INDIV.cds
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>229M<span class="w"> </span>May<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="m">10</span>:17<span class="w"> </span>Hadza_INDIV.gff
</pre></div>
</div>
<p>Hint: If you apply <code class="docutils literal notranslate"><span class="pre">prodigal</span></code> not <code class="docutils literal notranslate"><span class="pre">pyrodigal</span></code>, use this to fix gene ID in gff files by dbcan_utils (TIMING ~1min)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_utils<span class="w"> </span>gff_fix<span class="w"> </span>-i<span class="w"> </span>Hadza_INDIV_157.faa<span class="w"> </span>-g<span class="w"> </span>Hadza_INDIV_157.gff
</pre></div>
</div>
<p>The value attribution CDS ID in gff file and protein ID in fasta file generated by prodigal is not match, which will lead to failure in CGC prediction.
The command dbcan_utils is designed to fix this issue.</p>
</section>
</section>
</section>
<section id="module-2-run-dbcan-annotation-fig-2-to-obtain-cazymes-cgcs-and-substrates">
<h3>Module 2. run_dbcan annotation (Fig. 2) to obtain CAZymes, CGCs, and substrates<a class="headerlink" href="#module-2-run-dbcan-annotation-fig-2-to-obtain-cazymes-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<p><strong>CRITICAL STEP</strong></p>
<p>Users can skip P5 and P6, and directly run P7 (much slower though), if they want to predict not only CAZymes and CGCs, but also substrates.</p>
<section id="p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min">
<h4>P5. CAZyme annotation at the CAZyme family level (TIMING ~10min)<a class="headerlink" href="#p5-cazyme-annotation-at-the-cazyme-family-level-timing-10min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CAZyme<span class="w"> </span>--input_raw_data<span class="w"> </span>Hadza_INDIV_157.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>Hadza_INDIV_157.CAZyme<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>Two arguments are required for <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code>: the input sequence file (faa files) and the sequence type (protein).
By default, <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> will use three methods (<code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN</span> <span class="pre">HMMdb</span></code>, <code class="docutils literal notranslate"><span class="pre">DIAMOND</span></code> vs <code class="docutils literal notranslate"><span class="pre">CAZy</span></code>, <code class="docutils literal notranslate"><span class="pre">pyHMMER</span></code> vs <code class="docutils literal notranslate"><span class="pre">dbCAN-sub</span> <span class="pre">HMMdb</span></code>) for
CAZyme annotation.</p>
<p>The sequence type can be <cite>protein</cite>, <cite>prok</cite>, <cite>meta</cite>. If the input sequence file contains metagenomic contig sequences (<cite>fna</cite> file),
the sequence type has to be <cite>meta</cite>, and <cite>pyrodigal</cite> will be called to predict genes.</p>
</section>
<section id="p6-cgc-prediction-timing-15-min">
<h4>P6. CGC prediction (TIMING ~15 min)<a class="headerlink" href="#p6-cgc-prediction-timing-15-min" title="Permalink to this heading"></a></h4>
<p>The following commands will re-run run_dbcan to not only predict CAZymes but also CGCs with protein <cite>faa</cite> and gene location <cite>gff</cite> files.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_CGC<span class="w"> </span>--input_raw_data<span class="w"> </span>Hadza_INDIV_157.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--output_dir<span class="w"> </span>Hadza_INDIV_157.PUL<span class="w"> </span>--db_dir<span class="w"> </span>db<span class="w">  </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>Hadza_INDIV_157.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal
</pre></div>
</div>
<p>As mentioned above (see Fig. 1), CGC prediction is a featured function added into dbCAN2 in 2018.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p><strong>Creating own gff file</strong>
If the users would like to create their own <code class="docutils literal notranslate"><span class="pre">gff</span></code> file (instead of using Prokka or Prodigal),
it is important to make sure the value of ID attribute in the <code class="docutils literal notranslate"><span class="pre">gff</span></code> file matches the protein ID in the protein <code class="docutils literal notranslate"><span class="pre">faa</span></code> file.</p>
<p><strong>[Troubleshooting]CGC not found</strong>
If no result is found in CGC output file, it is most likely because the sequence IDs in <code class="docutils literal notranslate"><span class="pre">gff</span></code> file and <code class="docutils literal notranslate"><span class="pre">faa</span></code> file do not match.
Another less likely reason is that the contigs are too short and fragmented and not suitable for CGC prediction.</p>
</div>
</section>
<section id="p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h">
<h4>P7. Substrate prediction for CAZymes and CGCs (TIMING ~5h)<a class="headerlink" href="#p7-substrate-prediction-for-cazymes-and-cgcs-timing-5h" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>run_dbcan<span class="w"> </span>easy_substrate<span class="w"> </span>--input_raw_data<span class="w"> </span>Hadza_INDIV_157.faa<span class="w"> </span>--mode<span class="w"> </span>protein<span class="w"> </span>--input_format<span class="w"> </span>NCBI<span class="w"> </span>--input_gff<span class="w"> </span>Hadza_INDIV_157.gff<span class="w"> </span>--input_gff_format<span class="w"> </span>prodigal<span class="w"> </span>--output_dir<span class="w"> </span>Hadza_INDIV_157.dbCAN
</pre></div>
</div>
<section id="box-6-example-output-folder-content-of-run-dbcan-substrate-prediction">
<h5>Box 6. Example output folder content of run_dbcan substrate prediction<a class="headerlink" href="#box-6-example-output-folder-content-of-run-dbcan-substrate-prediction" title="Permalink to this heading"></a></h5>
<blockquote>
<div><p>In the output directory (<a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014.dbCAN/">Output Directory</a>),
a total of 17 files and 1 folder are generated:</p>
</div></blockquote>
<p>Descriptions of Output Files:
In the output directory, a total of 17 files and 1 folder are generated:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">PUL_blast.out</span></code>: BLAST results between CGCs and PULs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC.faa</span></code>: Protein Fasta sequences encoded in all CGCs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cgc.gff</span></code>: Reformatted from the user input gff file by marking CAZymes, TFs, TCs, and STPs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cgc.out</span></code>: Raw output of CGC predictions.</p></li>
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">cgc_standard.out</span></code>: Simplified version of cgc.out for easy parsing in TSV format. Example columns include:</dt><dd><ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_id</span></code>: CGC1</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">type</span></code>: CAZyme</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">contig_id</span></code>: k141_272079</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">gene_id</span></code>: k141_272079_6</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">start</span></code>: 5827</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">end</span></code>: 7257</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">strand</span></code>: -</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">annotation</span></code>: GH1</p></li>
</ol>
</dd>
</dl>
</li>
</ul>
<p><strong>Explanation</strong>: The gene Wet2014_00308 encodes a GH1 CAZyme in CGC1 of contig k141_272079. CGC1 also contains other genes, detailed in other rows. Wet2014_00308 is located on the negative strand of k141_272079 from positions 5827 to 7257. The type can be one of four signature gene types (CAZymes, TCs, TFs, STPs) or null type (not annotated as one of the signature genes).</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">dbCAN-sub.substrate.tsv</span></code>: HMMER search result against dbCAN-sub HMMdb, including a column with CAZyme substrates from fam-substrate-mapping.tsv.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">diamond_results.tsv</span></code>: DIAMOND search result against the CAZy annotated protein sequences (CAZyDB.fa).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">dbCAN_hmm_results.tsv</span></code>: HMMER search result against dbCAN HMMdb.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">overview.tsv</span></code>: Summary of CAZyme annotation from three methods in TSV format. Example columns include:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Gene_ID</span></code>: k141_355284_4</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">EC#</span></code>: 3.2.1.99:3</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">dbCAN</span></code>: GH43_4(42-453)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">dbCAN_sub</span></code>: GH43_e149</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">DIAMOND</span></code>: GH43_4</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">#ofTools</span></code>: 3</p></li>
</ol>
</div></blockquote>
</li>
</ul>
<p><strong>Explanation</strong>: The protein Wet2014_000076 is annotated by three tools as a CAZyme: GH43_4 (CAZy defined subfamily 4 of GH43) by HMMER vs dbCAN HMMdb, GH43_e149 (eCAMI defined subfamily e149; ‘e’ indicates it is from eCAMI not CAZy) by HMMER vs dbCAN-sub HMMdb, and GH43_4 by DIAMOND vs CAZy annotated protein sequences. The EC number is extracted from eCAMI, indicating that the eCAMI subfamily GH43_e149 contains 3 member proteins with an EC 3.2.1.99 according to CAZy. The preference order for different assignments is dbCAN &gt; eCAMI/dbCAN-sub &gt; DIAMOND. Refer to dbCAN2 paper11, dbCAN3 paper12, and eCAMI41 for more details.</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">stp.out</span></code>: HMMER search result against the MiST5 compiled signal transduction protein HMMs from Pfam.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">tf-1.out</span></code>: HMMER search result against the DBD6 compiled transcription factor HMMs from Pfam 7.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">tf-2.out</span></code>: HMMER search result against the DBD compiled transcription factor HMMs from Superfamily 8.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">tp.out</span></code>: DIAMOND search result against the TCDB 9 annotated protein sequences.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">substrate.out</span></code>: summary of substrate prediction results for CGCs in TSV format from two approaches3 (dbCAN-PUL blast search and dbCAN-sub majority voting). An example row has the following columns:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_ID</span></code>: k141_227425|CGC1</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Best</span> <span class="pre">hit</span> <span class="pre">PUL_ID</span> <span class="pre">in</span> <span class="pre">dbCAN-PUL</span></code>: PUL0402</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Substrate</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">hit</span> <span class="pre">PUL</span></code>: xylan</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Sum</span> <span class="pre">of</span> <span class="pre">bitscores</span> <span class="pre">for</span> <span class="pre">homologous</span> <span class="pre">gene</span> <span class="pre">pairs</span> <span class="pre">between</span> <span class="pre">CGC</span> <span class="pre">and</span> <span class="pre">PUL</span></code>: 2134.0</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Types</span> <span class="pre">of</span> <span class="pre">homologous</span> <span class="pre">gene</span> <span class="pre">pairs</span></code>: TC-TC;CAZyme-CAZyme</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Substrate</span> <span class="pre">predicted</span> <span class="pre">by</span> <span class="pre">majority</span> <span class="pre">voting</span> <span class="pre">of</span> <span class="pre">CAZymes</span> <span class="pre">in</span> <span class="pre">CGC</span></code>: xylan</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Voting</span> <span class="pre">score</span></code>: 2.0</p></li>
</ol>
</div></blockquote>
</li>
</ul>
<p><em>Explanation</em>: The CGC1 of contig k141_227425 has its best hit PUL0402 (from PUL_blast.out) with xylan as substrate (from dbCAN-PUL_12-12-2023.xlsx). Two signature genes are matched between k141_227425|CGC1 and PUL0402: one is a CAZyme and the other is a TC. The sum of blast bit scores of the two homologous pairs (TC-TC and CAZyme-CAZyme) is 2134.0. Hence, the substrate of k141_227425|CGC1 is predicted to be xylan according to dbCAN-PUL blast search. The last two columns are based on the dbCAN-sub result (dbcan-sub.hmm.out), as the file indicates that two CAZymes in k141_227425|CGC1 are predicted to have xylan substrate. The voting score is 2.0, so according to the majority voting rule, k141_227425|CGC1 is predicted to have a xylan substrate.</p>
<p><em>Note</em>: For many CGCs, only one of the two approaches produces substrate prediction. In some cases, the two approaches produce different substrate assignments. The recommended preference order is dbCAN-PUL blast search &gt; dbCAN-sub majority voting. Refer to dbCAN3 paper12 for more details.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">synteny.pdf</span></code>: A folder with syntenic block alignment plots between all CGCs and PULs.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">uniInput</span></code>: Renamed Fasta file from input protein sequence file.</p></li>
</ul>
</div></blockquote>
</section>
</section>
</section>
<section id="module-3-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates">
<h3>Module 3. Read mapping (Fig. 2) to calculate abundance for CAZyme families, subfamilies, CGCs, and substrates<a class="headerlink" href="#module-3-read-mapping-fig-2-to-calculate-abundance-for-cazyme-families-subfamilies-cgcs-and-substrates" title="Permalink to this heading"></a></h3>
<section id="p8-read-mapping-to-all-cds-of-each-sample-timing-20-min">
<h4>P8. Read mapping to all CDS of each sample (TIMING ~20 min)<a class="headerlink" href="#p8-read-mapping-to-all-cds-of-each-sample-timing-20-min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>samfiles
bwa<span class="w"> </span>index<span class="w"> </span>megahit_Hadza_INDIV_157/Hadza_INDIV_157.contigs.fa

bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/Dry2014.sam<span class="w"> </span>megahit_Hadza_INDIV_157/Hadza_INDIV_157.contigs.fa<span class="w"> </span>Dry2014_1_val_1.fq.gz<span class="w"> </span>Dry2014_2_val_2.fq.gz

bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>samfiles/Wet2014.sam<span class="w"> </span>megahit_Hadza_INDIV_157/Hadza_INDIV_157.contigs.fa<span class="w"> </span>Wet2014_1_val_1.fq.gz<span class="w"> </span>Wet2014_2_val_2.fq.gz
</pre></div>
</div>
<p>Reads are mapped to the <cite>contig</cite> files from MEGAHIT.</p>
</section>
<section id="p9-sort-sam-files-by-coordinates-timing-4min">
<h4>P9. Sort SAM files by coordinates (TIMING ~4min)<a class="headerlink" href="#p9-sort-sam-files-by-coordinates-timing-4min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>samfiles

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Wet2014.CDS.bam<span class="w"> </span>Wet2014.CDS.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Dry2014.CDS.bam<span class="w"> </span>Dry2014.CDS.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Wet2014.bam<span class="w"> </span>Wet2014.sam

samtools<span class="w"> </span>sort<span class="w"> </span>-@<span class="w"> </span><span class="m">32</span><span class="w"> </span>-o<span class="w"> </span>Dry2014.bam<span class="w"> </span>Dry2014.sam

samtools<span class="w"> </span>index<span class="w"> </span>Wet2014.bam

samtools<span class="w"> </span>index<span class="w"> </span>Dry2014.bam

rm<span class="w"> </span>-rf<span class="w"> </span>*sam

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>Reads are mapped to the <cite>contig</cite> files from MEGAHIT.</p>
</section>
<section id="p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-12min">
<h4>P10. Read count calculation for all proteins of each sample using dbcan_utils  (TIMING ~12min)<a class="headerlink" href="#p10-read-count-calculation-for-all-proteins-of-each-sample-using-dbcan-utils-timing-12min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>Wet2014_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>Wet2014_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../Hadza_INDIV_157.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/Wet2014.bam<span class="w"> </span>-o<span class="w"> </span>Wet2014.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>--identity<span class="w"> </span><span class="m">0</span>.98

<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>mkdir<span class="w"> </span>Dry2014_abund<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>Dry2014_abund

dbcan_utils<span class="w"> </span>cal_coverage<span class="w"> </span>-g<span class="w"> </span>../Hadza_INDIV_157.fix.gff<span class="w"> </span>-i<span class="w"> </span>../samfiles/Dry2014.bam<span class="w"> </span>-o<span class="w"> </span>Dry2014.depth.txt<span class="w"> </span>-t<span class="w"> </span><span class="m">6</span><span class="w"> </span>--overlap_base_ratio<span class="w"> </span><span class="m">0</span>.2<span class="w"> </span>--mapping_quality<span class="w"> </span><span class="m">30</span><span class="w"> </span>--identity<span class="w"> </span><span class="m">0</span>.98

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>How to filter the mapped reads are the key step of the abundance calculation. Several parameters can be set to filter the reads. The command dbcan_utils provides –overlap_base_ratio, –mapping_quality and –identity to filter the mapped reads which will may produce more reliable abundance. The default values for these three parameters are 0.2, 30 and 0.98, respectively. Read counts are saved in depth.txt files of each sample.</p>
</section>
<section id="p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-2min">
<h4>P11. Read count calculation for a given region of contigs using Samtools (TIMING ~2min)<a class="headerlink" href="#p11-read-count-calculation-for-a-given-region-of-contigs-using-samtools-timing-2min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>Wet2014_abund

samtools<span class="w"> </span>index<span class="w"> </span>../samfiles/Wet2014.bam

samtools<span class="w"> </span>depth<span class="w"> </span>-r<span class="w"> </span>k141_597014:160-8694<span class="w"> </span>../samfiles/Wet2014.bam<span class="w"> </span>&gt;<span class="w"> </span>Wet2014.cgc.depth.txt
</pre></div>
</div>
<p>The parameter -r k141_41392:152403-165349 specifies a region in a contig. For any CGC, its positional range can be found in the file cgc_standard.out produced by run_dbcan (Box 6).</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The contig IDs are automatically generated by MEGAHIT. There is a small chance that a same contig ID appears in both samples. However, the two contigs in the two samples do not match each other even the ID is the same. For example, the contig ID k141_4139 is most likely only found in the Wet2014 sample. Even if there is a k141_41392 in Dry2014, the actual contigs in two samples are different.
The depth.txt files contain the raw read counts for the specified region. It may show a warning but will not effect the result:
[W::hts_idx_load3] The index file is older than the data file: ../samfiles/Wet2014.bam.bai</p>
</div>
</section>
<section id="p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min">
<h4>P12. dbcan_utils to calculate the abundance of CAZyme families, subfamilies, CGCs, and substrates (TIMING ~1min)<a class="headerlink" href="#p12-dbcan-utils-to-calculate-the-abundance-of-cazyme-families-subfamilies-cgcs-and-substrates-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>Wet2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Wet2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>Wet2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Wet2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>Wet2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Wet2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>Wet2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Wet2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

<span class="nb">cd</span><span class="w"> </span>..<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>Dry2014_abund

dbcan_utils<span class="w"> </span>fam_abund<span class="w"> </span>-bt<span class="w"> </span>Dry2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Dry2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>fam_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>Dry2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Dry2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_abund<span class="w"> </span>-bt<span class="w"> </span>Dry2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Dry2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

dbcan_utils<span class="w"> </span>CGC_substrate_abund<span class="w"> </span>-bt<span class="w"> </span>Dry2014.depth.txt<span class="w"> </span>-i<span class="w"> </span>../Dry2014.dbCAN<span class="w"> </span>-a<span class="w"> </span>TPM

<span class="nb">cd</span><span class="w"> </span>..
</pre></div>
</div>
<p>We developed a set of Python scripts as dbcan_utils (included in the run_dbcan package) to take the raw read counts for all genes as input and output the normalized abundances (Box 7) of CAZyme families, subfamilies, CGCs, and substrates (Fig. 4). The parameter -a TPM can also be two other metrics: RPM, or RPKM61.</p>
<ul class="simple">
<li><p><strong>RPKM</strong> = # of mapped reads to a gene G / [(total # of mapped reads to all genes /106) x (gene G length/1000)]</p></li>
<li><p><strong>RPM</strong> = # of mapped reads to a gene G / (total # of mapped reads to all genes/106).</p></li>
<li><p><strong>TPM</strong> = [# of mapped reads to a gene G / (gene G length/1000)] / sum [# of mapped reads to each gene / (the gene length/1000)].</p></li>
</ul>
<section id="box-7-example-output-of-dbcan-utils">
<h5>Box 7. Example output of dbcan_utils<a class="headerlink" href="#box-7-example-output-of-dbcan-utils" title="Permalink to this heading"></a></h5>
<p>As an example, the Wet2014_abund folder (<a class="reference external" href="https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_abund/">https://bcb.unl.edu/dbCAN_tutorial/dataset1-Carter2023/individual_assembly/Wet2014_abund/</a>) has 7 TSV files:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span>226K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>CGC_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.3K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>CGC_substrate_majority_voting.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>20K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>CGC_substrate_PUL_homology.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">2</span>.8K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>EC_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w"> </span><span class="m">4</span>.1K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>fam_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>60K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>fam_substrate_abund.out
-rw-rw-r--<span class="w">  </span><span class="m">1</span><span class="w"> </span>jinfang<span class="w"> </span>jinfang<span class="w">  </span>30K<span class="w"> </span>Jun<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="m">09</span>:28<span class="w"> </span>subfam_abund.out
</pre></div>
</div>
<p>Explanation of columns in these TSV files is as follows:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">fam_abund.out</span></code>: CAZy family (from HMMER vs dbCAN HMMdb), sum of TPM, number of CAZymes in the family.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">subfam_abund.out</span></code>: eCAMI subfamily (from HMMER vs dbCAN-sub HMMdb), sum of TPM, number of CAZymes in the subfamily.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">EC_abund.out</span></code>: EC number (extracted from dbCAN-sub subfamily), sum of TPM, number of CAZymes with the EC.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fam_substrate_abund.out</span></code>: Substrate (from HMMER vs dbCAN-sub HMMdb), sum of TPM (all CAZymes in this substrate group), GeneID (all CAZyme IDs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_abund.out</span></code>: CGC_ID (e.g., k141_338400|CGC1), mean of TPM (all genes in the CGC), Seq_IDs (IDs of all genes in the CGC), TPM (of all genes in the CGC), Families (CAZyme family or other signature gene type of all genes in the CGC).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_PUL_homology.out</span></code>: Substrate (from dbCAN-PUL blast search), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-PUL blast search), TPM (of CGCs in this substrate group).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">CGC_substrate_majority_voting.out</span></code>: Substrate (from dbCAN-sub majority voting), sum of TPM, CGC_IDs (all CGCs predicted to have the substrate from dbCAN-sub majority voting), TPM (of CGCs in this substrate group).</p></li>
</ul>
</div></blockquote>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Proteins from multiple samples can be combined to generate a non-redundant set of proteins (Box 8). This may reduce the runtime for the run_dbcan step (step4), as only one faa file will be processed. However, this does not work for the CGC prediction, as contigs (fna files) from each sample will be needed. Therefore, this step is recommended if users only want the CAZyme annotation, and not recommended if CGCs are also to be predicted.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>As shown in Fig. 1 (step3), proteins from multiple samples can be combined to generate a non-redundant set of proteins (Box 8). This may reduce the runtime for the run_dbcan step (step4), as only one faa file will be processed. However, this does not work for the CGC prediction, as contigs (fna files) from each sample will be needed. Therefore, this step is recommended if users only want the CAZyme annotation, and not recommended if CGCs are also to be predicted.</p>
</div>
</section>
<section id="box-8-combine-proteins-from-multiple-samples-optional-timing-3h">
<h5>Box 8. Combine proteins from multiple samples (optional TIMING ~3h)<a class="headerlink" href="#box-8-combine-proteins-from-multiple-samples-optional-timing-3h" title="Permalink to this heading"></a></h5>
<p>This protein sequence clustering step will create a mapping table with sequence cluster ID and protein IDs from each sample.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>mmseqs_cluster<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>mmseqs_cluster
ln<span class="w"> </span>-s<span class="w"> </span>../db<span class="w"> </span>.
cat<span class="w"> </span>../Wet2014.faa<span class="w"> </span>../Dry2014.faa<span class="w"> </span>&gt;<span class="w"> </span>Dry_Wet.faa
mmseqs<span class="w"> </span>easy-cluster<span class="w"> </span>--threads<span class="w"> </span><span class="m">32</span><span class="w"> </span>-c<span class="w"> </span><span class="m">0</span>.95<span class="w"> </span>--min-seq-id<span class="w"> </span><span class="m">0</span>.95<span class="w"> </span>--cov-mode<span class="w"> </span><span class="m">2</span><span class="w"> </span>Dry_Wet.faa<span class="w"> </span>Dry_Wet_cluster<span class="w"> </span>tmp
mv<span class="w"> </span>Dry_Wet_cluster_cluster_rep.fasta<span class="w"> </span>Dry_Wet.cluster.faa
</pre></div>
</div>
<p>This Dry_Wet.cluster.faa file now contains the Fasta sequences of representative proteins of all mmseqs2 clusters, i.e., the non-redundant protein sequences from the two samples. The mapping table file Dry_Wet_cluster_cluster.tsv contains two columns, mmseqs2 cluster representative protein ID, and protein IDs in the cluster. Users can query this table to find the correspondence between CAZymes in Wet2014 and in Dry2014.</p>
</section>
</section>
</section>
<section id="module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable">
<h3>Module 4: dbcan_plot for data visualization (Fig. 2) of abundances of CAZymes, CGCs, and substrates (TIMING variable)<a class="headerlink" href="#module-4-dbcan-plot-for-data-visualization-fig-2-of-abundances-of-cazymes-cgcs-and-substrates-timing-variable" title="Permalink to this heading"></a></h3>
<p><strong>CRITICAL STEP</strong></p>
<p>To visualize the CAZyme annotation result, we provide a set of Python scripts as <code class="docutils literal notranslate"><span class="pre">dbcan_plot</span></code> to make publication-quality plots with the <code class="docutils literal notranslate"><span class="pre">dbcan_utils</span></code> results as the input. The <code class="docutils literal notranslate"><span class="pre">dbcan_plot</span></code> scripts are included in the <code class="docutils literal notranslate"><span class="pre">run_dbcan</span></code> package. Once the plots are made in PDF format, they can be transferred to users’ Windows or Mac computers for visualization.</p>
<p>Five data folders will be needed as the input for <code class="docutils literal notranslate"><span class="pre">dbcan_plot</span></code>:</p>
<ol class="arabic simple">
<li><p>Two abundance folders: <code class="docutils literal notranslate"><span class="pre">Wet2014_abund</span></code> and <code class="docutils literal notranslate"><span class="pre">Dry2014_abund</span></code>.</p></li>
<li><p>Two CAZyme annotation folders: <code class="docutils literal notranslate"><span class="pre">Wet2014.dbCAN</span></code> and <code class="docutils literal notranslate"><span class="pre">Dry2014.dbCAN</span></code>.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">dbCAN-PUL</span></code> folder (located under the db folder, released from <code class="docutils literal notranslate"><span class="pre">dbCAN-PUL.tar.gz</span></code>).</p></li>
</ol>
<section id="p13-heatmap-for-cazyme-substrate-abundance-across-samples-fig-6a-timing-1min">
<h4>P13. Heatmap for CAZyme substrate abundance across samples (Fig. 6A) (TIMING 1min)<a class="headerlink" href="#p13-heatmap-for-cazyme-substrate-abundance-across-samples-fig-6a-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>heatmap_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>-i<span class="w"> </span>Wet2014_abund/fam_substrate_abund.out,Dry2014_abund/fam_substrate_abund.out<span class="w"> </span>--show_abund<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span>
</pre></div>
</div>
<p>Here we plot the top 20 substrates in the two samples (Fig. 6A). The input files are the two CAZyme substrate abundance files calculated based on
dbCAN-sub result. The default heatmap is ranked by substrate abundances. To rank the heatmap according to abundance profile using
the clustermap function of the seaborn package (<a class="reference external" href="https://github.com/mwaskom/seaborn">https://github.com/mwaskom/seaborn</a>), users can invoke the <code class="docutils literal notranslate"><span class="pre">--cluster_map</span></code> parameter.</p>
</section>
<section id="p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-6b-c-timing-1min">
<h4>P14. Barplot for CAZyme family/subfamily/EC abundance across samples (Fig. 6B,C) (TIMING 1min)<a class="headerlink" href="#p14-barplot-for-cazyme-family-subfamily-ec-abundance-across-samples-fig-6b-c-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_abund/fam_abund.out,Dry2014_abund/fam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>fam_abund.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_abund/subfam_abund.out,Dry2014_abund/subfam_abund.out<span class="w"> </span>--pdf<span class="w"> </span>subfam_abund.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
dbcan_plot<span class="w"> </span>bar_plot<span class="w"> </span>--samples<span class="w"> </span>Wet2014,Dry2014<span class="w"> </span>--vertical_bar<span class="w"> </span>--top<span class="w"> </span><span class="m">20</span><span class="w"> </span>-i<span class="w"> </span>Wet2014_abund/EC_abund.out,Dry2014_abund/EC_abund.out<span class="w"> </span>--pdf<span class="w"> </span>EC_abund.pdf<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>Users can choose to generate a barplot instead of heatmap using the bar_plot method.</p>
</section>
<section id="p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-6d-timing-1min">
<h4>P15. Synteny plot between a CGC and its best PUL hit with read mapping coverage to CGC (Fig. 6D) (TIMING 1min)<a class="headerlink" href="#p15-synteny-plot-between-a-cgc-and-its-best-pul-hit-with-read-mapping-coverage-to-cgc-fig-6d-timing-1min" title="Permalink to this heading"></a></h4>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_coverage_plot<span class="w"> </span>-i<span class="w"> </span>Hadza_INDIV_157.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_41392|CGC3&#39;</span><span class="w"> </span>--readscount<span class="w"> </span>Wet2014_abund/Wet2014.cgc.depth.txt<span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<p>The Wet2014.dbCAN folder contains the PUL.out file. Using this file, the cgc_standard.out file, and the best PUL’s gff file in dbCAN-PUL.tar.gz, the CGC_synteny_plot method will create the CGC-PUL synteny plot. The –cgcid parameter is required to specify which CGC to plot (k141_41392|CGC3 in this example). The Wet2014.cgc.depth.txt file is used to plot the read mapping coverage.
If users only want to plot the CGC structure:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_plot<span class="w"> </span>-i<span class="w"> </span>Hadza_INDIV_157.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_41392|CGC3&#39;</span>
</pre></div>
</div>
<p>If users only want to plot the CGC structure plus the read mapping coverage:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_coverage_plot<span class="w"> </span>-i<span class="w"> </span>Hadza_INDIV_157.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_41392|CGC3&#39;</span><span class="w"> </span>--readscount<span class="w"> </span>Wet2014_abund/Wet2014.cgc.depth.txt
</pre></div>
</div>
<p>If users only want to plot the synteny between the CGC and PUL:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>dbcan_plot<span class="w"> </span>CGC_synteny_plot<span class="w"> </span>-i<span class="w"> </span>Hadza_INDIV_157.dbCAN<span class="w"> </span>--cgcid<span class="w"> </span><span class="s1">&#39;k141_41392|CGC3&#39;</span><span class="w"> </span>--db_dir<span class="w"> </span>db
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The CGC IDs in different samples do not match each other. For example, specifying -i Wet2014.dbCAN is to plot the ‘k141_41392|CGC3’ in the Wet2014 sample. The ‘k141_41392|CGC3’ in the Dry2014 sample most likely does not exist, and even it does, the CGC has a different sequence even if the ID is the same.</p>
</div>
</section>
</section>
</section>
<section id="troubleshooting">
<h2>Troubleshooting<a class="headerlink" href="#troubleshooting" title="Permalink to this heading"></a></h2>
<p>We provide Table 3 to list possible issues and solutions. Users can also post issues on run_dbcan GitHub site.</p>
</section>
<section id="procedure-timing">
<h2>Procedure Timing<a class="headerlink" href="#procedure-timing" title="Permalink to this heading"></a></h2>
<p>The estimated time for completing each step of the protocol on the Carter2023 dataset is as follows:</p>
<ul class="simple">
<li><p><strong>Step P1. Contamination checking ~10min.</strong></p></li>
<li><p><strong>Step P2. Raw reads processing ~20min.</strong></p></li>
<li><p><strong>Step P3. Metagenome assembly ~4h20min.</strong></p></li>
<li><p><strong>Step P4. Gene models prediction ~15min.</strong></p></li>
<li><p><strong>Step P5. CAZyme annotation ~10min.</strong></p></li>
<li><p><strong>Step P6. PUL prediction ~15min.</strong></p></li>
<li><p><strong>Step P7. Substrate prediction both for CAZyme and PUL ~5h.</strong></p></li>
<li><p><strong>Step P8-P11. Reads mapping ~36min.</strong></p></li>
<li><p><strong>Step P12. Abundance estimation ~1min.</strong></p></li>
<li><p><strong>Step P13-P15. Data visualization ~3min.</strong></p></li>
</ul>
<p>Running this protocol on the Carter2023 dataset will take ~12h on a Linux computer with 40 CPUs and 128GB of RAM. The most time-consuming step is P7 (substrate prediction for CGCs and CAZymes). .  If users choose not to predict substates, this step will take ~15min. RAM usage was not specifically monitored during the execution. The step with the highest RAM usage is likely P3 (read assembly).</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="run_from_raw_reads_em.html" class="btn btn-neutral float-left" title="Run from Raw Reads(Emilson 2024): Automated CAZyme and Glycan Substrate Annotation in Microbiomes: A Step-by-Step Protocol" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="run_from_raw_reads_sp_subsample.html" class="btn btn-neutral float-right" title="Run from Raw Reads(Cater 2023): Supplementary Protocol for subsample" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Xinpeng Zhang; Haidong Yi; Jinfang Zheng.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>